(this.webpackJsonpmosguito=this.webpackJsonpmosguito||[]).push([[0],{138:function(e,a,n){},164:function(e,a){},215:function(e,a,n){},224:function(e,a,n){"use strict";n.r(a);var t=n(2),i=n(0),o=n.n(i),s=n(12),r=n.n(s),c=(n(138),n(47)),l=n(88),m=n(13),d=n(87),u=n(16),p=n(61),h=n(34),b=n(92),g=n(51),f=n.n(g),O=n(89),j=n(259),y=n(263),x=n(77),C=(n(81),n(21)),S=n(116),v=n.n(S),E=n(28),I=function(e){var a=e.split("/");return(a=(a=a[a.length-1]).split("."))[0]},A=!1;function P(e){return R.apply(this,arguments)}function R(){return(R=Object(O.a)(f.a.mark((function e(a){var n,t,i,o,s,r,c,l,m,d,u,p,h,b,g,O,j,y,x,S,E,A,P,R,T,D,M,G,N,w,B,U,k,L,F;return f.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return n=a.target.files[0],t=new C.BlobReader(n),i=new C.ZipReader(t),e.next=5,i.getEntries();case 5:o=e.sent,s=[],r=[],c=[],l=[],m=[],d=[],u=[],p=[],h=[],b=[],g=0;case 17:if(!(g<o.length)){e.next=82;break}if(!1!==o[g].directory||0==o[g].compressedSize){e.next=79;break}if(!o[g].filename.includes("Preprocess")){e.next=25;break}return e.next=22,o[g].getData(new C.BlobWriter(["text/html"]));case 22:O=e.sent,j=I(o[g].filename),s.push({name:j,blob:O});case 25:if(!o[g].filename.includes("Annotation")){e.next=31;break}return e.next=28,o[g].getData(new C.BlobWriter(["text/html"]));case 28:y=e.sent,x=I(o[g].filename),r.push({name:x,blob:y});case 31:if(!o[g].filename.includes("Differential expression analysis")){e.next=37;break}return e.next=34,o[g].getData(new C.BlobWriter(["image/jpeg"]));case 34:S=e.sent,E=I(o[g].filename),c.push({name:E,blob:S});case 37:if(!o[g].filename.includes("KEGGMaps")){e.next=44;break}return e.next=40,o[g].getData(new C.BlobWriter(["image/png"]));case 40:A=e.sent,P=I(o[g].filename),R=l.length,l.push({name:[R,P],blob:A});case 44:if(!o[g].filename.includes("Assembly")){e.next=50;break}return e.next=47,o[g].getData(new C.BlobWriter(["text/tab-separated-values"]));case 47:T=e.sent,D=I(o[g].filename),m.push({name:D,blob:T});case 50:if(!o[g].filename.includes("Entry")){e.next=56;break}return e.next=53,o[g].getData(new C.BlobWriter(["text/tab-separated-values"]));case 53:M=e.sent,G=I(o[g].filename),d.push({name:G,blob:M});case 56:if(!o[g].filename.includes("config")){e.next=62;break}return e.next=59,o[g].getData(new C.BlobWriter(["\tapplication/json"]));case 59:N=e.sent,w=URL.createObjectURL(N),v.a.getJSON(w,(function(e){u=e}));case 62:if(!o[g].filename.includes("experiments")){e.next=67;break}return e.next=65,o[g].getData(new C.BlobWriter(["text/tab-separated-values"]));case 65:B=e.sent,p=B;case 67:if(!o[g].filename.includes("General")){e.next=73;break}return e.next=70,o[g].getData(new C.BlobWriter(["text/tab-separated-values"]));case 70:U=e.sent,k=I(o[g].filename),h.push({name:k,blob:U});case 73:if(!o[g].filename.includes("Protein")){e.next=79;break}return e.next=76,o[g].getData(new C.BlobWriter(["text/tab-separated-values"]));case 76:L=e.sent,F=I(o[g].filename),b.push({name:F,blob:L});case 79:g++,e.next=17;break;case 82:return e.next=84,i.close();case 84:return e.abrupt("return",[{qcReports:s,KronaPlots:r,Heatmaps:c,KEGGMaps:l,asReports:m,entryReport:d,generalReport:h,proteinReport:b},u,p]);case 85:case"end":return e.stop()}}),e)})))).apply(this,arguments)}var T=function(e){e.outputsFiles;var a=e.setOutputsFiles,n=e.onConfigOverwrite,i=e.setExperiments,o=e.setExperimentsRows,s=function(){var e=Object(O.a)(f.a.mark((function e(t){var s,r,l;return f.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return e.next=2,P(t);case 2:s=e.sent,a(s[0]),Object.keys(s[1]).map((function(e){return delete Object.assign(s[1],Object(c.a)({},(a=e,a.replace(/([-_][a-z])/gi,(function(e){return e.toUpperCase().replace("-","").replace("_","")}))),s[1][e]))[e];var a})),n(s[1]),r=function(e){E.parse(e,{download:!0,header:!0,complete:function(e){e.data.pop();var a=e.data;i(a),o(Object.keys(a).length)}})},l=URL.createObjectURL(s[2]),r(l);case 9:case"end":return e.stop()}}),e)})));return function(a){return e.apply(this,arguments)}}();return Object(t.jsx)(t.Fragment,{children:Object(t.jsxs)(j.a,{variant:"contained",color:"secondary",component:"label",onClick:function(){A=!0},children:["Select results' ZIP archive",Object(t.jsx)("input",{type:"file",accept:"application/zip",onChange:s,hidden:!0})]})})},D=function(e){var a=e.outputsFiles,n=e.setOutputsFiles,i=e.onConfigOverwrite,o=e.setExperiments,s=e.setExperimentsRows;return Object(t.jsxs)(N,{children:[Object(t.jsx)(y.a,{children:Object(t.jsx)(x.a,{variant:"h6",children:"MOSCA results page"})}),Object(t.jsx)(T,{outputsFiles:a,setOutputsFiles:n,onConfigOverwrite:i,setExperiments:o,setExperimentsRows:s})]})},M=(n(213),function(e){e.outputsFiles;var a=Object(u.f)(),n=Object(u.g)(),s=Object(i.useState)(!1),r=Object(m.a)(s,2),c=r[0],l=r[1];return!1===A?Object(t.jsxs)(o.a.Fragment,{children:[Object(t.jsx)("div",{onClick:function(){return l(!1)},className:"fixed inset-0 z-20 block transition-opacity bg-black opacity-50 lg:hidden ".concat(c?"block":"hidden")}),Object(t.jsx)("div",{children:Object(t.jsx)("button",{className:"btn-menu",onClick:function(){return l(!c)},type:"button",children:Object(t.jsx)(h.a,{name:"burger",className:"w-6 h-6"})})}),Object(t.jsxs)("div",{className:"fixed inset-y-0 left-0 z-30 w-64 overflow-y-auto transition duration-300 ease-out transform translate-x-0 bg-white border-r-2 lg:translate-x-0 lg:static lg:inset-0 ".concat(c?"ease-out -translate-x-0":"ease-in -translate-x-full"),children:[Object(t.jsx)(p.Navigation,{activeItemId:n.pathname,onSelect:function(e){var n=e.itemId;a.push(n)},items:[{title:"Home",itemId:"/MOSGUITO/home",elemBefore:function(){return Object(t.jsx)(h.a,{name:"coffee"})}},{title:"About",itemId:"/MOSGUITO/about",elemBefore:function(){return Object(t.jsx)(h.a,{name:"user"})},subNav:[{title:"Project",itemId:"/MOSGUITO/project"},{title:"Members",itemId:"/MOSGUITO/members"}]},{title:"Configuration",itemId:"/MOSGUITO/config",elemBefore:function(){return Object(t.jsx)(h.a,{name:"settings"})},subNav:[{title:"General configuration",itemId:"/MOSGUITO/general-configuration"},{title:"Experiments",itemId:"/MOSGUITO/experiments"},{title:"UniProt columns",itemId:"/MOSGUITO/uniprot-columns"},{title:"UniProt databases",itemId:"/MOSGUITO/uniprot-databases"},{title:"KEGG metabolic maps",itemId:"/MOSGUITO/keggmaps"},{title:"Proteomics configuration",itemId:"/MOSGUITO/proteomics-configuration"}]}]}),Object(t.jsx)("div",{className:"absolute bottom-0 w-full my-8",children:Object(t.jsx)(p.Navigation,{activeItemId:n.pathname,items:[{title:"Results",itemId:"/MOSGUITO/results",elemBefore:function(){return Object(t.jsx)(b.a,{})}}],onSelect:function(e){var n=e.itemId;a.push(n)}})})]})]}):Object(t.jsxs)(o.a.Fragment,{children:[Object(t.jsx)("div",{onClick:function(){return l(!1)},className:"fixed inset-0 z-20 block transition-opacity bg-black opacity-50 lg:hidden ".concat(c?"block":"hidden")}),Object(t.jsx)("div",{children:Object(t.jsx)("button",{className:"btn-menu",onClick:function(){return l(!c)},type:"button",children:Object(t.jsx)(h.a,{name:"burger",className:"w-6 h-6"})})}),Object(t.jsxs)("div",{className:"fixed inset-y-0 left-0 z-30 w-64 overflow-y-auto transition duration-300 ease-out transform translate-x-0 bg-white border-r-2 lg:translate-x-0 lg:static lg:inset-0 ".concat(c?"ease-out -translate-x-0":"ease-in -translate-x-full"),children:[Object(t.jsx)(p.Navigation,{activeItemId:n.pathname,onSelect:function(e){var n=e.itemId;a.push(n)},items:[{title:"Home",itemId:"/MOSGUITO/home",elemBefore:function(){return Object(t.jsx)(h.a,{name:"coffee"})}},{title:"About",itemId:"/MOSGUITO/about",elemBefore:function(){return Object(t.jsx)(h.a,{name:"user"})},subNav:[{title:"Project",itemId:"/MOSGUITO/project"},{title:"Members",itemId:"/MOSGUITO/members"}]},{title:"Configuration",itemId:"/MOSGUITO/config",elemBefore:function(){return Object(t.jsx)(h.a,{name:"settings"})},subNav:[{title:"General configuration",itemId:"/MOSGUITO/general-configuration"},{title:"Experiments",itemId:"/MOSGUITO/experiments"},{title:"UniProt columns",itemId:"/MOSGUITO/uniprot-columns"},{title:"UniProt databases",itemId:"/MOSGUITO/uniprot-databases"},{title:"KEGG metabolic maps",itemId:"/MOSGUITO/keggmaps"},{title:"Proteomics configuration",itemId:"/MOSGUITO/proteomics-configuration"}]}]}),Object(t.jsx)("div",{className:"absolute bottom-0 w-full my-8",children:Object(t.jsx)(p.Navigation,{activeItemId:n.pathname,items:[{title:"Results",itemId:"/MOSGUITO/results",elemBefore:function(){return Object(t.jsx)(b.a,{})},subNav:[{title:"Load results",itemId:"/MOSGUITO/load-results"},{title:"FastQC reports",itemId:"/MOSGUITO/fastqc-reports"},{title:"Assembly QC",itemId:"/MOSGUITO/assembly-qc"},{title:"Annotation Results",itemId:"/MOSGUITO/annotation-results"},{title:"Differential Analysis",itemId:"/MOSFUITO/differential-analysis"},{title:"KEGGmaps",itemId:"/MOSGUITO/keggmaps-results"},{title:"EntryReports",itemId:"/MOSGUITO/entry-reports"},{title:"GeneralReports",itemId:"/MOSGUITO/general-reports"},{title:"ProteinReports",itemId:"/MOSGUITO/protein-reports"}]}],onSelect:function(e){var n=e.itemId;a.push(n)}})})]})]})}),G=function(e){var a=e.children;return Object(t.jsx)("div",{className:"relative min-h-screen",children:Object(t.jsx)("main",{className:"w-full min-h-screen",children:a})})},N=function(e){var a=e.children;return Object(t.jsx)(G,{children:Object(t.jsxs)("div",{className:"flex h-screen bg-gray-200 flex-grow-1",children:[Object(t.jsx)(M,{}),Object(t.jsx)("div",{className:"flex flex-col flex-1 overflow-hidden flex-grow-1",children:Object(t.jsx)("main",{className:"content w-100 m-r-1-rem",children:Object(t.jsx)("section",{className:"sm:flex-row flex flex-col flex-1",children:Object(t.jsx)("div",{className:"content-box",style:{flexGrow:2,flexBasis:"0%"},children:a})})})})]})})},w=function(){return Object(t.jsx)(N,{children:Object(t.jsx)("h2",{children:"Project Page"})})},B=n(264),U=n(267),k=n(268),L=n(226),F=n(265),H=n(286),K=n(266),Y=Object(B.a)((function(e){return{root:{width:"100%",maxWidth:"36ch",backgroundColor:e.palette.background.paper},inline:{display:"inline"}}})),z=function(e){var a=e.name,n=e.imageLink,i=e.pageLink,s=e.company,r=e.role,c=Y();return Object(t.jsxs)(L.a,{alignItems:"flex-start",children:[Object(t.jsx)(F.a,{children:Object(t.jsx)(H.a,{alt:a,src:n})}),Object(t.jsx)(K.a,{primary:a,secondary:Object(t.jsxs)(o.a.Fragment,{children:[Object(t.jsx)(x.a,{component:"span",variant:"body2",className:c.inline,color:"textPrimary",children:r}),Object(t.jsx)("a",{href:i,target:"_blank",rel:"noreferrer",children:Object(t.jsxs)("span",{style:{color:"orange"},children:["at ",s]})})]})})]})},q=Object(B.a)((function(e){return{root:{width:"100%",maxWidth:"36ch",backgroundColor:e.palette.background.paper},inline:{display:"inline"}}}));function V(){var e=q();return Object(t.jsxs)(U.a,{className:e.root,children:[Object(t.jsx)(z,{name:"Andreia F. Salvador",imageLink:"https://www.ceb.uminho.pt/Media/GetMedia/c7b7a6ab-a690-4dc7-8703-a055dc9a4ac2",pageLink:"https://www.ceb.uminho.pt/People/Details/c6b2fd64-389d-4700-ae5e-7ba9b7f26321",company:"CEB",role:""}),Object(t.jsx)(k.a,{variant:"inset",component:"li"}),Object(t.jsx)(z,{name:"Jo\xe3o C. Sequeira",imageLink:"https://www.ceb.uminho.pt/Media/GetMedia/4ce02006-2ae9-4173-9b48-12b1617d733f",pageLink:"https://www.ceb.uminho.pt/People/Details/9fca8b19-e2c7-4d5f-acb8-c7e15c98ca47",company:"CEB",role:""}),Object(t.jsx)(k.a,{variant:"inset",component:"li"}),Object(t.jsx)(z,{name:"M. Madalena Alves",imageLink:"https://www.ceb.uminho.pt/Media/GetMedia/67e1e08f-794f-4a6b-ae2e-c5928f429bee",pageLink:"https://www.ceb.uminho.pt/People/Details/44265aed-6255-48b0-965d-8dbecf383f6e",company:"CEB",role:""})]})}var W=function(){return Object(t.jsx)(N,{children:Object(t.jsx)(V,{})})},_=function(){return Object(t.jsx)(N,{})},Q=function(){return Object(t.jsx)(N,{children:Object(t.jsx)("h2",{children:"Teams Page"})})},X=function(){return Object(t.jsx)("div",{className:"App",children:Object(t.jsxs)("header",{className:"header",children:[Object(t.jsx)(x.a,{variant:"h4",children:"MOSGUITO"}),Object(t.jsx)(x.a,{variant:"h6",children:"MOSca's GUI TO generate config files"})]})})},J=function(){return Object(t.jsx)(N,{children:Object(t.jsx)(X,{})})},Z=n(25),$=n(117),ee=n.n($),ae=n(274),ne=n(275),te=n(276),ie=n(283),oe=function(e){var a=e.label,n=e.value,i=e.onChange,o=e.placeholder;return Object(t.jsxs)("div",{style:{margin:"0.5rem 0"},children:[Object(t.jsx)("span",{children:a}),Object(t.jsx)(ie.a,{type:"text",fullWidth:!0,value:n,onChange:i,placeholder:o})]})},se=function(e){var a=e.label,n=e.value,i=e.onChange;return Object(t.jsxs)("div",{style:{margin:"0.5rem 0"},children:[Object(t.jsx)("span",{children:a}),Object(t.jsx)(ie.a,{type:"number",fullWidth:!0,value:n,onChange:i,inputProps:{min:1}})]})},re=n(284),ce=function(e){var a=e.label,n=e.checked,i=e.setChecked;return Object(t.jsxs)("div",{children:[Object(t.jsx)(re.a,{checked:n,onChange:i}),Object(t.jsx)("span",{children:a})]})},le=n(282),me=n(271),de=function(e){var a=e.label,n=e.value,i=e.onChange,o=e.options;return Object(t.jsxs)("div",{style:{margin:"0.5rem 0"},children:[Object(t.jsxs)("span",{children:[a,"\t"]}),Object(t.jsx)(le.a,{value:n,onChange:i,children:o.map((function(e){return Object(t.jsx)(me.a,{value:e,children:e},e)}))})]})},ue=["metaspades","megahit"],pe=["illumina_5","illumina_10","sanger_5","sanger_10","454_10","454_30","complete"],he=["CDD","Pfam","NCBIfam","Protein_Clusters","Smart","TIGRFAM","COG","KOG"],be=["40","107"],ge=["TMM","RLE"],fe=["SUPERKINGDOM","PHYLUM","CLASS","ORDER","FAMILY","GENUS","SPECIES"],Oe=["compomics","maxquant"],je=["SUPERKINGDOM","PHYLUM","CLASS","ORDER","FAMILY","GENUS","SPECIES"],ye=["Trypsin","File"],xe={output:"output",resourcesDirectory:"resources_directory",threads:14,experiments:"experiments.tsv",minimumReadLength:100,minimumReadAverageQuality:20,doAssembly:!0,maxMemory:4,assembler:ue[0],errorModel:pe[0],markerset:be[0],diamondDatabase:"resources_directory/uniprot.dmnd",diamondMaxTargetSeqs:1,downloadUniprot:!0,downloadCdd:!0,uniprotColumns:["id","entry name","genes","protein names","ec","comment(FUNCTION)","comment(PATHWAY)","keywords","existence","go","families","lineage(SUPERKINGDOM)","lineage(PHYLUM)","lineage(CLASS)","lineage(ORDER)","lineage(FAMILY)","lineage(GENUS)","lineage(SPECIES)"],uniprotDatabases:["BioCyc","BRENDA","CDD","eggNOG","Ensembl","InterPro","KEGG","KO","Pfam","Reactome","RefSeq","UniPathway","GO"],recognizerDatabases:he,normalizationMethod:ge[0],keggcharterMaps:["00010","00020","00030","00040","00051","00052","00053","00500","00520","00620","00630","00640","00650","00660","00562","00190","00195","00196","00710","00720","00680","00910","00920","00061","00062","00071","00072","00073","00100","00120","00121","00140","00561","00564","00565","00600","00590","00591","00592","01040","00230","00240","00250","00260","00270","00280","00290","00310","00220","00330","00340","00350","00360","00380","00400","00410","00430","00440","00450","00460","00471","00472","00473","00480","00510","00513","00512","00515","00514","00532","00534","00533","00531","00563","00601","00603","00604","00540","00550","00511","00571","00572","00730","00740","00750","00760","00770","00780","00785","00790","00670","00830","00860","00130","00900","00902","00909","00904","00906","00905","00981","00908","00903","00281","01052","00522","01051","01059","01056","01057","00253","00523","01054","01053","01055","00940","00945","00941","00944","00942","00943","00901","00403","00950","00960","01058","00232","00965","00966","00402","00311","00332","00261","00331","00521","00524","00525","00401","00404","00405","00333","00254","00998","00999","00362","00627","00364","00625","00361","00623","00622","00633","00642","00643","00791","00930","00363","00621","00626","00624","00365","00984","00980","03020","03022","03040","03010","00970","03013","03015","03008","03060","04141","04130","04120","04122","03050","03018","03030","03410","03420","03430","03440","03450","03460","02010","02060","03070","02020","04014","04015","04010","04011","04012","04310","04330","04340","04350","04390","04392","04370","04371","04630","04064","04668","04066","04068","04020","04070","04072","04071","04024","04022","04151","04152","04150","04075","04080","04060","04061","04512","04514","04144","04145","04142","04146","04138","04136","04139","04110","04111","04112","04113","04114","04210","04215","04216","04217","04115","04218","04510","04520","04530","04540","04550","02024","05111","02025","02026","02030","02040","04810"],keggcharterTaxaLevel:fe[0],keggcharterNumberOfTaxa:10,proteomicsWorkflow:"compomics",useCrap:!0,proteomicsContaminantesDatabase:"crap.fasta",referenceProteomesTaxaLevel:"GENUS",protease:"Trypsin",proteaseFile:""},Ce={output:"",resourcesDirectory:"",threads:1,experiments:"",doAssembly:!0,assembler:ue[0],errorModel:pe[0],markerset:be[0],diamondDatabase:"",diamondMaxTargetSeqs:1,downloadUniprot:!0,uniprotColumns:[],uniprotDatabases:[],recognizerDatabases:[],normalizationMethod:ge[0],keggcharterMaps:[],keggcharterTaxaLevel:fe[0],keggcharterNumberOfTaxa:1};var Se=function(e,a,n){var t=new Blob([e],{type:n});if(window.navigator.msSaveOrOpenBlob)window.navigator.msSaveOrOpenBlob(t,a);else{var i=document.createElement("a"),o=URL.createObjectURL(t);i.href=o,i.download=a,document.body.appendChild(i),i.click(),setTimeout((function(){document.body.removeChild(i),window.URL.revokeObjectURL(o)}),0)}},ve=(n(215),n(285)),Ee=n(272),Ie=n(273),Ae=n(118),Pe=n.n(Ae),Re=function(e){var a=e.title,n=e.style,o=e.children,s=Object(i.useState)(!1),r=Object(m.a)(s,2),c=r[0],l=r[1];return Object(t.jsxs)(ve.a,{expanded:c,onChange:function(){return l(!c)},style:n||{width:"100%"},children:[Object(t.jsx)(Ee.a,{expandIcon:Object(t.jsx)(Pe.a,{}),children:a}),Object(t.jsx)(Ie.a,{children:o})]})},Te=function(e){var a=e.configData,n=e.onConfigChange,i=e.onConfigOverwrite,o=function(e){return e.replace(/[A-Z]/g,(function(e){return"_".concat(e.toLowerCase())}))};return Object(t.jsx)("main",{className:"main",children:Object(t.jsx)("form",{className:"form",children:Object(t.jsxs)(ae.a,{children:[Object(t.jsxs)(ne.a,{children:[Object(t.jsxs)(x.a,{variant:"body1",children:["To run MOSCA, you need both a configuration and an experiments files. MOSGUITO is used to obtained configuration files, and you can ",Object(t.jsx)("a",{href:"https://docs.google.com/spreadsheets/d/12BppOf32QPRl6Ey39ACWoOXPVWtTrGIUN2u-_BpwSvo/edit#gid=0",children:Object(t.jsx)("span",{style:{color:"red"},children:"access"})})," an example of experiments file from MOSCA's repository."]}),Object(t.jsx)(k.a,{style:{margin:"1rem 0"}}),Object(t.jsx)(oe,{label:"Output directory",value:a.output,onChange:function(e){return n("output",e.target.value)},placeholder:xe.output}),Object(t.jsx)(oe,{label:"Resources directory",value:a.resourcesDirectory,onChange:function(e){return n("resourcesDirectory",e.target.value)},placeholder:xe.resourcesDirectory}),Object(t.jsx)(oe,{label:"Experiments filename",value:a.experiments,onChange:function(e){return n("experiments",e.target.value)},placeholder:xe.experiments}),Object(t.jsx)(se,{label:"Number of threads to use",value:a.threads,onChange:function(e){return n("threads",e.target.valueAsNumber)}}),Object(t.jsx)(se,{label:"Minimum read length",value:a.minimumReadLength,onChange:function(e){return n("minimumReadLength",e.target.valueAsNumber)}}),Object(t.jsx)(se,{label:"Minimum read average quality",value:a.minimumReadAverageQuality,onChange:function(e){return n("minimumReadAverageQuality",e.target.valueAsNumber)}}),Object(t.jsx)(se,{label:"Maximum memory (Gb)",value:a.maxMemory,onChange:function(e){return n("maxMemory",e.target.valueAsNumber)}}),Object(t.jsx)(ce,{label:"Perform assembly",checked:a.doAssembly,setChecked:function(e){return n("doAssembly",e.target.checked)}}),a.doAssembly?Object(t.jsxs)(t.Fragment,{children:[Object(t.jsx)(de,{label:"Choose assembler",value:a.assembler,onChange:function(e){return n("assembler",e.target.value)},options:ue}),Object(t.jsx)(de,{label:"Choose markerset",value:a.markerset,onChange:function(e){return n("markerset",e.target.value)},options:be})]}):Object(t.jsx)(de,{label:"Choose error model",value:a.errorModel,onChange:function(e){return n("errorModel",e.target.value)},options:pe}),Object(t.jsx)(oe,{label:"DIAMOND database",value:a.diamondDatabase,onChange:function(e){return n("diamondDatabase",e.target.value)},placeholder:xe.diamondDatabase}),Object(t.jsx)(ce,{label:"Download UniProt",checked:a.downloadUniprot,setChecked:function(e){return n("downloadUniprot",e.target.checked)}}),Object(t.jsx)(ce,{label:"Download CDD",checked:a.downloadCdd,setChecked:function(e){return n("downloadCdd",e.target.checked)}}),Object(t.jsx)(Re,{title:"Pick databases of reCOGnizer",children:he.map((function(e,i){return Object(t.jsx)(ce,{label:e,checked:a.recognizerDatabases.indexOf(e)>-1,setChecked:function(t){return function(e){var t=Object(Z.a)(a.recognizerDatabases),i=t.indexOf(e);i>-1?t.splice(i,1):t.push(e),n("recognizerDatabases",t)}(e)}},i)}))}),Object(t.jsx)(se,{label:"Number of identifications per protein",value:a.diamondMaxTargetSeqs,onChange:function(e){return n("diamondMaxTargetSeqs",e.target.valueAsNumber)}}),Object(t.jsx)(de,{label:"Choose normalization method",value:a.normalizationMethod,onChange:function(e){return n("normalizationMethod",e.target.value)},options:ge}),Object(t.jsx)(de,{label:"Choose the taxonomic level to represent with KEGGCharter",value:a.keggcharterTaxaLevel,onChange:function(e){return n("keggcharterTaxaLevel",e.target.value)},options:fe}),Object(t.jsx)(se,{label:"Number of taxa to represent with KEGGCharter",value:a.keggcharterNumberOfTaxa,onChange:function(e){return n("keggcharterNumberOfTaxa",e.target.valueAsNumber)}})]}),Object(t.jsxs)(te.a,{style:{display:"flex",justifyContent:"flex-end"},children:[Object(t.jsx)(j.a,{onClick:function(){return i(xe)},variant:"contained",color:"primary",children:"Set to default values"}),Object(t.jsx)(j.a,{onClick:function(){return i(Ce)},variant:"contained",color:"primary",children:"Clear values"}),Object(t.jsx)(j.a,{onClick:function(e){return function(e){e.preventDefault(),a.doAssembly&&n("errorModel","complete");var t={};Object.keys(a).map((function(e){return t[o(e)]=a[e]})),Se(ee.a.stringify(t,null),"config.yaml","yaml")}(e)},variant:"contained",color:"secondary",children:"Download YAML"}),Object(t.jsx)(j.a,{onClick:function(e){return function(e){e.preventDefault(),a.doAssembly&&n("errorModel","complete");var t={};Object.keys(a).map((function(e){return t[o(e)]=a[e]})),Se(JSON.stringify(t,null,2),"config.json","json")}(e)},variant:"contained",color:"secondary",children:"Download JSON"})]})]})})})},De=function(){return Object(t.jsxs)("header",{className:"header",children:[Object(t.jsx)(x.a,{variant:"h4",children:"General configuration"}),Object(t.jsx)(x.a,{variant:"h6",children:"Tweak different parameters to customize MOSCA's workflow"})]})};var Me=function(e){var a=e.configData,n=e.onConfigChange,i=e.onConfigOverwrite;return Object(t.jsx)(N,{children:Object(t.jsxs)("div",{className:"App",children:[Object(t.jsx)(De,{}),Object(t.jsx)(Te,{configData:a,onConfigChange:n,onConfigOverwrite:i})]})})},Ge=n(7),Ne=n(227),we=n(277),Be=n(278),Ue=n(279),ke=n(280),Le=n(281),Fe=Object(Ge.a)((function(e){return{root:{width:"100%",marginTop:e.spacing(3),overflowX:"auto"},table:{minWidth:700}}}))((function(e){var a=e.experiments,n=e.setExperiments,i=e.nExperimentsRows,o=e.setExperimentsRows,s=function(e,t,i){var o=Object(Z.a)(a);o[t][i]=e,n(o)};return Object(t.jsxs)(Ne.a,{children:[Object(t.jsx)(y.a,{children:Object(t.jsx)("div",{children:Object(t.jsx)(x.a,{variant:"h6",children:"Experiments"})})}),Object(t.jsxs)(we.a,{children:[Object(t.jsx)(Be.a,{children:Object(t.jsxs)(Ue.a,{children:[Object(t.jsx)(ke.a,{children:"Files"}),Object(t.jsx)(ke.a,{width:"20%",children:"Sample"}),Object(t.jsx)(ke.a,{width:"15%",children:"Data type"}),Object(t.jsx)(ke.a,{width:"10%",children:"Condition"}),Object(t.jsx)(ke.a,{width:"20%",children:"Name"})]})}),Object(t.jsx)(Le.a,{children:Array(i).fill().map((function(e,n){return Object(t.jsxs)(Ue.a,{children:[Object(t.jsx)(ke.a,{children:Object(t.jsx)(ie.a,{type:"text",value:a[n].Files,onChange:function(e){return s(e.target.value,n,"Files")},placeholder:"",multiline:!0,fullWidth:!0})}),Object(t.jsx)(ke.a,{children:Object(t.jsx)(ie.a,{type:"text",value:a[n].Sample,onChange:function(e){return s(e.target.value,n,"Sample")},placeholder:""})}),Object(t.jsx)(ke.a,{children:Object(t.jsx)(de,{type:"text",value:a[n]["Data type"],onChange:function(e){return s(e.target.value,n,"Data type")},options:["dna","mrna","protein"]})}),Object(t.jsx)(ke.a,{children:Object(t.jsx)(ie.a,{type:"text",value:a[n].Condition,onChange:function(e){return s(e.target.value,n,"Condition")},placeholder:""})}),Object(t.jsx)(ke.a,{children:Object(t.jsx)(ie.a,{type:"text",value:a[n].Name,onChange:function(e){return s(e.target.value,n,"Name")},placeholder:""})})]},n)}))})]}),Object(t.jsx)(j.a,{onClick:function(e){return function(){var e=Object(Z.a)(a);e.push({Files:"",Sample:"","Data type":"",Condition:"",Name:""}),n(e),o(i+1)}()},variant:"contained",color:"primary",children:"Add row"}),Object(t.jsx)(j.a,{onClick:function(e){return function(){var e=Object(Z.a)(a);e.pop(),n(e),o(i-1)}()},variant:"contained",color:"primary",children:"Remove last row"})]})})),He=n(119),Ke=n.n(He),Ye=function(e){var a=e.experiments,n=e.setExperiments,i=e.nExperimentsRows,o=e.setExperimentsRows;return Object(t.jsxs)(N,{children:[Object(t.jsx)(Fe,{experiments:a,setExperiments:n,nExperimentsRows:i,setExperimentsRows:o}),Object(t.jsx)(j.a,{onClick:function(e){return Se(Ke.a.stringify(a),"experiments.tsv","tsv")},variant:"contained",color:"secondary",children:"Download TSV"})]})},ze=function(e){var a=e.uniprotList,n=e.onChange,i=e.uniprotPossibilities;return Object.entries(i).map((function(e,i){var o=Object(m.a)(e,2),s=o[0],r=o[1];return Object(t.jsx)(Re,{title:s,children:r.map((function(e,i){return Object(t.jsx)(ce,{label:e[0],checked:a.indexOf(e[1])>-1,setChecked:function(){return function(e){var t=Object(Z.a)(a),i=t.indexOf(e);i>-1?t.splice(i,1):t.push(e),n(t)}(e[1])}},i)}))},i)}))},qe={"Names & Taxonomy":[["Entry","id"],["Entry name","entry name"],["Gene names","genes"],["Gene names (primary)","genes(PREFERRED)"],["Gene names (synonym)","genes(ALTERNATIVE)"],["Gene names (ordered locus)","genes(OLN)"],["Gene names (ORF)","genes(ORF)"],["Organism","organism"],["Organism ID","organism-id"],["Protein names","protein names"],["Proteomes","proteome"],["Taxonomic lineage","lineage(ALL)"],["Virus hosts","virus hosts"]],Sequences:[["Fragment","fragment"],["Gene encoded by","encodedon"],["Alternative products","comment(ALTERNATIVE PRODUCTS)"],["Erroneous gene model prediction","comment(ERRONEOUS GENE MODEL PREDICTION)"],["Erroneous initiation","comment(ERRONEOUS INITIATION)"],["Erroneous termination","comment(ERRONEOUS TERMINATION)"],["Erroneous translation","comment(ERRONEOUS TRANSLATION)"],["Frameshift","comment(FRAMESHIFT)"],["Mass spectrometry","comment(MASS SPECTROMETRY)"],["Polymorphism","comment(POLYMORPHISM)"],["RNA editing","comment(RNA EDITING)"],["Sequence caution","comment(SEQUENCE CAUTION)"],["Length","length"],["Mass","mass"],["Sequence","sequence"],["Alternative sequence","feature(ALTERNATIVE SEQUENCE)"],["Natural variant","feature(NATURAL VARIANT)"],["Non-adjacent residues","feature(NON ADJACENT RESIDUES)"],["Non-standard residue","feature(NON STANDARD RESIDUE)"],["Non-terminal residue","feature(NON TERMINAL RESIDUE)"],["Sequence conflict","feature(SEQUENCE CONFLICT)"],["Sequence uncertainty","feature(SEQUENCE UNCERTAINTY)"],["Sequence version","version(sequence)"]],Function:[["Absorption","comment(ABSORPTION)"],["Active site","feature(ACTIVE SITE)"],["Binding site","feature(BINDING SITE)"],["Catalytic activity","comment(CATALYTIC ACTIVITY)"],["ChEBI","chebi"],["ChEBI (Catalytic activity)","chebi(Catalytic activity)"],["ChEBI (Cofactor)","chebi(Cofactor)"],["ChEBI IDs","chebi-id"],["Cofactor","comment(COFACTOR)"],["DNA binding","feature(DNA BINDING)"],["EC number","ec"],["Enzyme regulation","comment(ENZYME REGULATION)"],["Function[CC]","comment(FUNCTION)"],["Kinetics","comment(KINETICS)"],["Metal binding","feature(METAL BINDING)"],["Nucleotide binding","feature(NP BIND)"],["Pathway","comment(PATHWAY)"],["pH dependence","comment(PH DEPENDENCE)"],["Redox potential","comment(REDOX POTENTIAL)"],["Rhea ids","rhea-id"],["Site","feature(SITE)"],["Temperature dependence","comment(TEMPERATURE DEPENDENCE)"]],Miscellaneous:[["Annotation score","annotation score"],["Features","features"],["Caution","comment(CAUTION)"],["Miscellaneous[CC]","comment(MISCELLANEOUS)"],["Keywords","keywords"],["Matched text","context"],["Protein existence","existence"],["Tools","tools"],["Reviewed","reviewed"]],Interaction:[["Subunit structure[CC]","comment(SUBUNIT)"],["Interacts with","interactor"]],Expression:[["Developmental stage","comment(DEVELOPMENTAL STAGE)"],["Induction","comment(INDUCTION)"],["Tissue specificity","comment(TISSUE SPECIFICITY)"]],"Gene Ontology (GO)":[["Gene ontology (GO)","go"],["Gene ontology (biological process)","go(biological process)"],["Gene ontology (molecular function)","go(molecular function)"],["Gene ontology (cellular component)","go(cellular component)"],["Gene ontology IDs","go-id"]],"Pathology & Biotech":[["Allergenic properties","comment(ALLERGEN)"],["Biotechnological use","comment(BIOTECHNOLOGY)"],["Disruption phenotype","comment(DISRUPTION PHENOTYPE)"],["Involvement in disease","comment(DISEASE)"],["Pharmaceutical use","comment(PHARMACEUTICAL)"],["Toxic dose","comment(TOXIC DOSE)"]],"Subcellular location":[["Subcellular location[CC]","comment(SUBCELLULAR LOCATION)"],["Intramembrane","feature(INTRAMEMBRANE)"],["Topological domain","feature(TOPOLOGICAL DOMAIN)"],["Transmembrane","feature(TRANSMEMBRANE)"]],"PTM / Processsing":[["Post-translational modification","comment(PTM)"],["Chain","feature(CHAIN)"],["Cross-link","feature(CROSS LINK)"],["Disulfide bond","feature(DISULFIDE BOND)"],["Glycosylation","feature(GLYCOSYLATION)"],["Initiator methionine","feature(INITIATOR METHIONINE)"],["Lipidation","feature(LIPIDATION)"],["Modified residue","feature(MODIFIED RESIDUE)"],["Peptide","feature(PEPTIDE)"],["Propeptide","feature(PROPEPTIDE)"],["Signal peptide","feature(SIGNAL)"],["Transit peptide","feature(TRANSIT)"]],Structure:[["3D","3d"],["Beta strand","feature(BETA STRAND)"],["Helix","feature(HELIX)"],["Turn","feature(TURN)"]],Publications:[["Mapped PubMed ID","citationmapping"],["PubMed ID","citation"]],"Date of":[["Date of creation","created"],["Date of last modification","last-modified"],["Date of last sequence modification","sequence-modified"],["Entry version","version(entry)"]],"Family & Domains":[["Domain[CC]","comment(DOMAIN)"],["Sequence similarities","comment(SIMILARITY)"],["Protein families","families"],["Coiled coil","feature(COILED COIL)"],["Compositional bias","feature(COMPOSITIONAL BIAS)"],["Domain[FT]","feature(DOMAIN EXTENT)"],["Motif","feature(MOTIF)"],["Region","feature(REGION)"],["Repeat","feature(REPEAT)"],["Zinc finger","feature(ZINC FINGER)"]],"Taxonomic lineage":[["Taxonomic lineage (all)","lineage(all)"],["Taxonomic lineage (SUPERKINGDOM)","lineage(SUPERKINGDOM)"],["Taxonomic lineage (KINGDOM)","lineage(KINGDOM)"],["Taxonomic lineage (SUBKINGDOM)","lineage(SUBKINGDOM)"],["Taxonomic lineage (SUPERPHYLUM)","lineage(SUPERPHYLUM)"],["Taxonomic lineage (PHYLUM)","lineage(PHYLUM)"],["Taxonomic lineage (SUBPHYLUM)","lineage(SUBPHYLUM)"],["Taxonomic lineage (SUPERCLASS)","lineage(SUPERCLASS)"],["Taxonomic lineage (CLASS)","lineage(CLASS)"],["Taxonomic lineage (SUBCLASS)","lineage(SUBCLASS)"],["Taxonomic lineage (INFRACLASS)","lineage(INFRACLASS)"],["Taxonomic lineage (SUPERORDER)","lineage(SUPERORDER)"],["Taxonomic lineage (ORDER)","lineage(ORDER)"],["Taxonomic lineage (SUBORDER)","lineage(SUBORDER)"],["Taxonomic lineage (INFRAORDER)","lineage(INFRAORDER)"],["Taxonomic lineage (PARVORDER)","lineage(PARVORDER)"],["Taxonomic lineage (SUPERFAMILY)","lineage(SUPERFAMILY)"],["Taxonomic lineage (FAMILY)","lineage(FAMILY)"],["Taxonomic lineage (SUBFAMILY)","lineage(SUBFAMILY)"],["Taxonomic lineage (TRIBE)","lineage(TRIBE)"],["Taxonomic lineage (SUBTRIBE)","lineage(SUBTRIBE)"],["Taxonomic lineage (GENUS)","lineage(GENUS)"],["Taxonomic lineage (SUBGENUS)","lineage(SUBGENUS)"],["Taxonomic lineage (SPECIES GROUP)","lineage(SPECIES GROUP)"],["Taxonomic lineage (SPECIES SUBGROUP)","lineage(SPECIES SUBGROUP)"],["Taxonomic lineage (SPECIES)","lineage(SPECIES)"],["Taxonomic lineage (SUBSPECIES)","lineage(SUBSPECIES)"],["Taxonomic lineage (VARIETAS)","lineage(VARIETAS)"],["Taxonomic lineage (FORMA)","lineage(FORMA)"]],"Taxonomic identifier":[["Taxonomic identifier (all)","lineage-id(all)"],["Taxonomic identifier (SUPERKINGDOM)","lineage-id(SUPERKINGDOM)"],["Taxonomic identifier (KINGDOM)","lineage-id(KINGDOM)"],["Taxonomic identifier (SUBKINGDOM)","lineage-id(SUBKINGDOM)"],["Taxonomic identifier (SUPERPHYLUM)","lineage-id(SUPERPHYLUM)"],["Taxonomic identifier (PHYLUM)","lineage-id(PHYLUM)"],["Taxonomic identifier (SUBPHYLUM)","lineage-id(SUBPHYLUM)"],["Taxonomic identifier (SUPERCLASS)","lineage-id(SUPERCLASS)"],["Taxonomic identifier (CLASS)","lineage-id(CLASS)"],["Taxonomic identifier (SUBCLASS)","lineage-id(SUBCLASS)"],["Taxonomic identifier (INFRACLASS)","lineage-id(INFRACLASS)"],["Taxonomic identifier (SUPERORDER)","lineage-id(SUPERORDER)"],["Taxonomic identifier (ORDER)","lineage-id(ORDER)"],["Taxonomic identifier (SUBORDER)","lineage-id(SUBORDER)"],["Taxonomic identifier (INFRAORDER)","lineage-id(INFRAORDER)"],["Taxonomic identifier (PARVORDER)","lineage-id(PARVORDER)"],["Taxonomic identifier (SUPERFAMILY)","lineage-id(SUPERFAMILY)"],["Taxonomic identifier (FAMILY)","lineage-id(FAMILY)"],["Taxonomic identifier (SUBFAMILY)","lineage-id(SUBFAMILY)"],["Taxonomic identifier (TRIBE)","lineage-id(TRIBE)"],["Taxonomic identifier (SUBTRIBE)","lineage-id(SUBTRIBE)"],["Taxonomic identifier (GENUS)","lineage-id(GENUS)"],["Taxonomic identifier (SUBGENUS)","lineage-id(SUBGENUS)"],["Taxonomic identifier (SPECIES GROUP)","lineage-id(SPECIES GROUP)"],["Taxonomic identifier (SPECIES SUBGROUP)","lineage-id(SPECIES SUBGROUP)"],["Taxonomic identifier (SPECIES)","lineage-id(SPECIES)"],["Taxonomic identifier (SUBSPECIES)","lineage-id(SUBSPECIES)"],["Taxonomic identifier (VARIETAS)","lineage-id(VARIETAS)"],["Taxonomic identifier (FORMA)","lineage-id(FORMA)"]]},Ve=function(e){var a=e.uniprotList,n=e.onChange;return Object(t.jsxs)("main",{className:"main",children:[Object(t.jsx)(y.a,{children:Object(t.jsx)("div",{children:Object(t.jsx)(x.a,{variant:"h6",children:"UniProt columns"})})}),Object(t.jsx)(ze,{uniprotList:a,onChange:n,uniprotPossibilities:qe})]})},We=function(e){var a=e.uniprotList,n=e.onChange;return Object(t.jsx)(N,{children:Object(t.jsx)(Ve,{uniprotList:a,onChange:n})})},_e={"Protocols and materials databases":[["ABCD curated depository of sequenced antibodies","ABCD"],["Antibodypedia a portal for validated antibodies","Antibodypedia"],["The CPTC Antibody Portal","CPTC"],["The DNASU plasmid repository","DNASU"]],"Protein family/group databases":[["Allergome; a platform for allergen knowledge","Allergome"],["Carbohydrate-Active enZymes","CAZy"],["CLAE, a database of fungal genes encoding lignocellulose-active proteins","CLAE"],["ESTHER database of the Alpha/Beta-hydrolase fold superfamily of proteins","ESTHER"],["Information system for G protein-coupled receptors (GPCRs)","GPCRDB"],["The international ImMunoGeneTics information system","IMGT_GENE-DB"],["MEROPS protease database","MEROPS"],["MoonDB Database of extreme multifunctional and moonlighting proteins","MoonDB"],["MoonProt database of moonlighting proteins","MoonProt"],["PeroxiBase, a peroxidase database","PeroxiBase"],["Restriction enzymes and methylases database","REBASE"],["Transport Classification Database","TCDB"],["UniLectin database of carbohydrate-binding proteins","UniLectin"]],"Organism-specific databases":[["ArachnoServer: Spider toxin database","ArachnoServer"],["Arabidopsis Information Portal","Araport"],["Candida Genome Database","CGD"],["ConoServer: Cone snail toxin database","ConoServer"],["Comparative Toxicogenomics Database","CTD"],["Dictyostelium discoideum online informatics resource","dictyBase"],["DisGeNET","DisGeNET"],["EchoBASE - an integrated post-genomic database for E. coli","EchoBASE"],["European Hepatitis C Virus Database","euHCVdb"],["Eukaryotic Pathogen Database Resources","EuPathDB"],["Drosophila genome database","FlyBase"],["GenAtlas: human gene database","GenAtlas"],["GeneCards: human genes, protein and diseases","GeneCards"],["GeneReviews a resource of expert-authored, peer-reviewed disease descriptions.","GeneReviews"],["Human Gene Nomenclature Database","HGNC"],["Human Protein Atlas","HPA"],["Human Unidentified Gene-Encoded large proteins database","HUGE"],["Legionella pneumophila genome database","LegioList"],["Mycobacterium leprae genome database","Leproma"],["Maize Genetics and Genomics Database","MaizeGDB"],["MalaCards human disease database","MalaCards"],["Mouse genome database (MGD) from Mouse Genome Informatics (MGI)","MGI"],["Microbial advanced database","Micado"],["Online Mendelian Inheritance in Man (OMIM)","MIM"],["neXtProt; the human protein knowledge platform","neXtProt"],["NIAGADS Genomics Database","NIAGADS"],["Open Targets","OpenTargets"],["Orphanet; a database dedicated to information on rare diseases and orphan drugs","Orphanet"],["The Pharmacogenetics and Pharmacogenomics Knowledge Base","PharmGKB"],["Schizosaccharomyces pombe database","PomBase"],["Pseudomonas genome database","PseudoCAP"],["Rat genome database","RGD"],["Rodent Unidentified Gene-Encoded large proteins database","Rouge"],["Saccharomyces Genome Database","SGD"],["The Arabidopsis Information Resource","TAIR"],["Mycobacterium tuberculosis strain H37Rv genome database","TubercuList"],["Vertebrate Gene Nomenclature Database","VGNC"],["WormBase","WormBase"],["Xenopus laevis and tropicalis biology and genomics resource","Xenbase"],["Zebrafish Information Network genome database","ZFIN"]],"Gene expression databases":[["Bgee dataBase for Gene Expression Evolution","Bgee"],["CollecTF database of bacterial transcription factor binding sites","CollecTF"],["ExpressionAtlas, Differential and Baseline Expression","ExpressionAtlas"],["Genevisible search portal to normalized and curated expression data from Genevestigator","Genevisible"]],"Chemistry databases":[["BindingDB database of measured binding affinities","BindingDB"],["ChEMBL database of bioactive drug-like small molecules","ChEMBL"],["Drug and drug target database","DrugBank"],["DrugCentral","DrugCentral"],["IUPHAR/BPS Guide to PHARMACOLOGY","GuidetoPHARMACOLOGY"],["SwissLipids knowledge resource for lipid biology","SwissLipids"]],"Enzyme and pathway databases":[["BioCyc Collection of Pathway/Genome Databases","BioCyc"],["BRENDA Comprehensive Enzyme Information System","BRENDA"],["Enzyme nomenclature database","ENZYME"],["Pathway Commons web resource for biological pathway data","PathwayCommons"],["Reactome - a knowledgebase of biological pathways and processes for plant species","PlantReactome"],["Reactome - a knowledgebase of biological pathways and processes","Reactome"],["SABIO-RK: Biochemical Reaction Kinetics Database","SABIO-RK"],["SignaLink: a signaling pathway resource with multi-layered regulatory networks","SignaLink"],["SIGNOR Signaling Network Open Resource","SIGNOR"],["UniPathway: a resource for the exploration and annotation of metabolic pathways","UniPathway"]],"Protein-protein interaction databases":[["The Biological General Repository for Interaction Datasets (BioGRID)","BioGRID"],["ComplexPortal: manually curated resource of macromolecular complexes","ComplexPortal"],["CORUM comprehensive resource of mammalian protein complexes","CORUM"],["Database of interacting proteins","DIP"],["The Eukaryotic Linear Motif resource for Functional Sites in Proteins","ELM"],["Protein interaction database and analysis system","IntAct"],["Molecular INTeraction database","MINT"],["STRING: functional protein association networks","STRING"]],"Miscellaneous databases":[["BioGRID ORCS database of CRISPR phenotype screens","BioGRID-ORCS"],["ChiTaRS: a database of human, mouse and fruit fly chimeric transcripts and RNA-sequencing data","ChiTaRS"],["Relative evolutionary importance of amino acids within a protein sequence","EvolutionaryTrace"],["The Gene Wiki collection of pages on human genes and proteins","GeneWiki"],["Database of phenotypes from RNA interference screens in Drosophila and Homo sapiens","GenomeRNAi"],["Pharos NIH Druggable Genome Knowledgebase","Pharos"],["Pathogen-Host Interaction database","PHI-base"],["Protein Ontology","PRO"],["Proteomes","Proteomes"],["RNAct, Protein-RNA interaction predictions for model organisms.","RNAct"],["The Stanford Online Universal Resource for Clones and ESTs","SOURCE"]],"Polymorphism and mutation databases":[["BioMuta curated single-nucleotide variation and disease association database","BioMuta"],["Database of single nucleotide polymorphism","dbSNP"],["Domain mapping of disease mutations (DMDM)","DMDM"]],"3D structure databases":[["Biological Magnetic Resonance Data Bank","BMRB"],["Database of comparative protein structure models","ModBase"],["The Protein Circular Dichroism Data Bank","PCDDB"],["Protein Data Bank Europe","PDB"],["Protein Data Bank in Europe - Knowledge Base","PDBe-KB"],["Protein Data Bank Japan","PDBj"],["PDBsum; at-a-glance overview of macromolecular structures","PDBsum"],["Protein Data Bank RCSB","RCSB-PDB"],["Small Angle Scattering Biological Data Bank","SASBDB"],["SWISS-MODEL Repository - a database of annotated 3D protein structure models","SMR"],["SWISS-MODEL Interactive Workspace","SWISS-MODEL-Workspace"]],"PTM databases":[["CarbonylDB database of protein carbonylation sites","CarbonylDB"],["DEPOD human dephosphorylation database","DEPOD"],["GlyConnect protein glycosylation platform","GlyConnect"],["GlyGen: Computational and Informatics Resources for Glycoscience","GlyGen"],["iPTMnet integrated resource for PTMs in systems biology context","iPTMnet"],["MetOSite database of methionine sulfoxide sites","MetOSite"],["Comprehensive resource for the study of protein post-translational modifications (PTMs) in human, mouse and rat.","PhosphoSitePlus"],["SwissPalm database of S-palmitoylation events","SwissPalm"],["UniCarbKB; an annotated and curated database of glycan structures","UniCarbKB"]],"Sequence databases":[["The Consensus CDS (CCDS) project","CCDS"],["DNA Data Bank of Japan; a nucleotide sequence database","DDBJ"],["EMBL nucleotide sequence database","EMBL"],["GenBank nucleotide sequence database","GenBank"],["Protein sequence database of the Protein Information Resource","PIR"],["NCBI Reference Sequences","RefSeq"]],"Family and domain databases":[["Conserved Domains Database","CDD"],["Database of protein disorder","DisProt"],["Gene3D Structural and Functional Annotation of Protein Families","Gene3D"],["HAMAP database of protein families","HAMAP"],["Intrinsically Disordered proteins with Extensive Annotations and Literature","IDEAL"],["Integrated resource of protein families, domains and functional sites","InterPro"],["MobiDB: a database of protein disorder and mobility annotations","MobiDB"],["The PANTHER Classification System","PANTHER"],["Pfam protein domain database","Pfam"],["PIRSF; a whole-protein classification database","PIRSF"],["Protein Motif fingerprint database; a protein domain database","PRINTS"],["PROSITE; a protein domain and family database","PROSITE"],["ProtoNet; Automatic hierarchical classification of proteins","ProtoNet"],["Structure-Function Linkage Database","SFLD"],["Simple Modular Architecture Research Tool; a protein domain database","SMART"],["Superfamily database of structural and functional annotation","SUPFAM"],["TIGRFAMs; a protein family database","TIGRFAMs"]],"2D gel databases":[["2-DE database at Universidad Complutense de Madrid","COMPLUYEAST-2DPAGE"],["DOSAC-COBS 2D-PAGE database","DOSAC-COBS-2DPAGE"],["USC-OGP 2-DE database","OGP"],["REPRODUCTION-2DPAGE","REPRODUCTION-2DPAGE"],["Two-dimensional polyacrylamide gel electrophoresis database from the Geneva University Hospital","SWISS-2DPAGE"],["University College Dublin 2-DE Proteome Database","UCD-2DPAGE"],["The World-2DPAGE database","World-2DPAGE"]],"Proteomic databases":[["The CPTAC Assay portal","CPTAC"],["Encyclopedia of Proteome Dynamics","EPD"],["jPOST - Japan Proteome Standard Repository/Database","jPOST"],["MassIVE - Mass Spectrometry Interactive Virtual Environment","MassIVE"],["MaxQB - The MaxQuant DataBase","MaxQB"],["PaxDb, a database of protein abundance averages across all three domains of life","PaxDb"],["PeptideAtlas","PeptideAtlas"],["PRoteomics IDEntifications database","PRIDE"],["Protein Mass spectra EXtraction","ProMEX"],["ProteomicsDB: a multi-organism proteome resource","ProteomicsDB"],["Consortium for Top Down Proteomics","TopDownProteomics"]],"Phylogenomic databases":[["evolutionary genealogy of genes: Non-supervised Orthologous Groups","eggNOG"],["Ensembl GeneTree","GeneTree"],["The HOGENOM Database of Homologous Genes from Fully Sequenced Organisms","HOGENOM"],["InParanoid: Eukaryotic Ortholog Groups","InParanoid"],["KEGG Orthology (KO)","KO"],["Identification of Orthologs from Complete Genome Data","OMA"],["Database of Orthologous Groups","OrthoDB"],["Database for complete collections of gene phylogenies","PhylomeDB"],["TreeFam database of animal gene trees","TreeFam"]],"Genome annotation databases":[["Ensembl eukaryotic genome annotation project","Ensembl"],["Ensembl bacterial and archaeal genome annotation project","EnsemblBacteria"],["Ensembl fungal genome annotation project","EnsemblFungi"],["Ensembl metazoan genome annotation project","EnsemblMetazoa"],["Ensembl plant genome annotation project","EnsemblPlants"],["Ensembl protists genome annotation project","EnsemblProtists"],["GeneDB pathogen genome database from Sanger Institute","GeneDB"],["Database of genes from NCBI RefSeq genomes","GeneID"],["Gramene; a comparative resource for plants","Gramene"],["KEGG: Kyoto Encyclopedia of Genes and Genomes","KEGG"],["Pathosystems Resource Integration Center (PATRIC)","PATRIC"],["UCSC genome browser","UCSC"],["Bioinformatics Resource for Invertebrate Vectors of Human Pathogens","VectorBase"],["WormBase ParaSite","WBParaSite"]],Ontologies:[["Gene Ontology","GO"]]},Qe=function(e){var a=e.uniprotList,n=e.onChange;return Object(t.jsxs)("main",{className:"main",children:[Object(t.jsx)(y.a,{children:Object(t.jsx)("div",{children:Object(t.jsx)(x.a,{variant:"h6",children:"UniProt databases"})})}),Object(t.jsx)(ze,{uniprotList:a,onChange:n,uniprotPossibilities:_e})]})},Xe=function(e){var a=e.uniprotList,n=e.onChange;return Object(t.jsx)(N,{children:Object(t.jsx)(Qe,{uniprotList:a,onChange:n})})},Je={name:"br08901",children:[{name:"Metabolism",children:[{name:"Global and overview maps",children:[{name:["01100","Metabolic pathways"]},{name:["01110","Biosynthesis of secondary metabolites"]},{name:["01120","Microbial metabolism in diverse environments"]},{name:["01200","Carbon metabolism"]},{name:["01210","2-Oxocarboxylic acid metabolism"]},{name:["01212","Fatty acid metabolism"]},{name:["01230","Biosynthesis of amino acids"]},{name:["01220","Degradation of aromatic compounds"]}]},{name:"Carbohydrate metabolism",children:[{name:["00010","Glycolysis | Gluconeogenesis"]},{name:["00020","Citrate cycle (TCA cycle)"]},{name:["00030","Pentose phosphate pathway"]},{name:["00040","Pentose and glucuronate interconversions"]},{name:["00051","Fructose and mannose metabolism"]},{name:["00052","Galactose metabolism"]},{name:["00053","Ascorbate and aldarate metabolism"]},{name:["00500","Starch and sucrose metabolism"]},{name:["00520","Amino sugar and nucleotide sugar metabolism"]},{name:["00620","Pyruvate metabolism"]},{name:["00630","Glyoxylate and dicarboxylate metabolism"]},{name:["00640","Propanoate metabolism"]},{name:["00650","Butanoate metabolism"]},{name:["00660","C5-Branched dibasic acid metabolism"]},{name:["00562","Inositol phosphate metabolism"]}]},{name:"Energy metabolism",children:[{name:["00190","Oxidative phosphorylation"]},{name:["00195","Photosynthesis"]},{name:["00196","Photosynthesis - antenna proteins"]},{name:["00710","Carbon fixation in photosynthetic organisms"]},{name:["00720","Carbon fixation pathways in prokaryotes"]},{name:["00680","Methane metabolism"]},{name:["00910","Nitrogen metabolism"]},{name:["00920","Sulfur metabolism"]}]},{name:"Lipid metabolism",children:[{name:["00061","Fatty acid biosynthesis"]},{name:["00062","Fatty acid elongation"]},{name:["00071","Fatty acid degradation"]},{name:["00072","Synthesis and degradation of ketone bodies"]},{name:["00073","Cutin, suberine and wax biosynthesis"]},{name:["00100","Steroid biosynthesis"]},{name:["00120","Primary bile acid biosynthesis"]},{name:["00121","Secondary bile acid biosynthesis"]},{name:["00140","Steroid hormone biosynthesis"]},{name:["00561","Glycerolipid metabolism"]},{name:["00564","Glycerophospholipid metabolism"]},{name:["00565","Ether lipid metabolism"]},{name:["00600","Sphingolipid metabolism"]},{name:["00590","Arachidonic acid metabolism"]},{name:["00591","Linoleic acid metabolism"]},{name:["00592","alpha-Linolenic acid metabolism"]},{name:["01040","Biosynthesis of unsaturated fatty acids"]}]},{name:"Nucleotide metabolism",children:[{name:["00230","Purine metabolism"]},{name:["00240","Pyrimidine metabolism"]}]},{name:"Amino acid metabolism",children:[{name:["00250","Alanine, aspartate and glutamate metabolism"]},{name:["00260","Glycine, serine and threonine metabolism"]},{name:["00270","Cysteine and methionine metabolism"]},{name:["00280","Valine, leucine and isoleucine degradation"]},{name:["00290","Valine, leucine and isoleucine biosynthesis"]},{name:["00300","Lysine biosynthesis"]},{name:["00310","Lysine degradation"]},{name:["00220","Arginine biosynthesis"]},{name:["00330","Arginine and proline metabolism"]},{name:["00340","Histidine metabolism"]},{name:["00350","Tyrosine metabolism"]},{name:["00360","Phenylalanine metabolism"]},{name:["00380","Tryptophan metabolism"]},{name:["00400","Phenylalanine, tyrosine and tryptophan biosynthesis"]}]},{name:"Metabolism of other amino acids",children:[{name:["00410","beta-Alanine metabolism"]},{name:["00430","Taurine and hypotaurine metabolism"]},{name:["00440","Phosphonate and phosphinate metabolism"]},{name:["00450","Selenocompound metabolism"]},{name:["00460","Cyanoamino acid metabolism"]},{name:["00471","D-Glutamine and D-glutamate metabolism"]},{name:["00472","D-Arginine and D-ornithine metabolism"]},{name:["00473","D-Alanine metabolism"]},{name:["00480","Glutathione metabolism"]}]},{name:"Glycan biosynthesis and metabolism",children:[{name:["00510","N-Glycan biosynthesis"]},{name:["00513","Various types of N-glycan biosynthesis"]},{name:["00512","Mucin type O-glycan biosynthesis"]},{name:["00515","Mannose type O-glycan biosynthesis"]},{name:["00514","Other types of O-glycan biosynthesis"]},{name:["00532","Glycosaminoglycan biosynthesis - chondroitin sulfate | dermatan sulfate"]},{name:["00534","Glycosaminoglycan biosynthesis - heparan sulfate | heparin"]},{name:["00533","Glycosaminoglycan biosynthesis - keratan sulfate"]},{name:["00531","Glycosaminoglycan degradation"]},{name:["00563","Glycosylphosphatidylinositol (GPI)-anchor biosynthesis"]},{name:["00601","Glycosphingolipid biosynthesis - lacto and neolacto series"]},{name:["00603","Glycosphingolipid biosynthesis - globo and isoglobo series"]},{name:["00604","Glycosphingolipid biosynthesis - ganglio series"]},{name:["00540","Lipopolysaccharide biosynthesis"]},{name:["00541","O-Antigen nucleotide sugar biosynthesis"]},{name:["00550","Peptidoglycan biosynthesis"]},{name:["00511","Other glycan degradation"]},{name:["00571","Lipoarabinomannan (LAM) biosynthesis"]},{name:["00572","Arabinogalactan biosynthesis - Mycobacterium"]}]},{name:"Metabolism of cofactors and vitamins",children:[{name:["00730","Thiamine metabolism"]},{name:["00740","Riboflavin metabolism"]},{name:["00750","Vitamin B6 metabolism"]},{name:["00760","Nicotinate and nicotinamide metabolism"]},{name:["00770","Pantothenate and CoA biosynthesis"]},{name:["00780","Biotin metabolism"]},{name:["00785","Lipoic acid metabolism"]},{name:["00790","Folate biosynthesis"]},{name:["00670","One carbon pool by folate"]},{name:["00830","Retinol metabolism"]},{name:["00860","Porphyrin and chlorophyll metabolism"]},{name:["00130","Ubiquinone and other terpenoid-quinone biosynthesis"]}]},{name:"Metabolism of terpenoids and polyketides",children:[{name:["00900","Terpenoid backbone biosynthesis"]},{name:["00902","Monoterpenoid biosynthesis"]},{name:["00909","Sesquiterpenoid and triterpenoid biosynthesis"]},{name:["00904","Diterpenoid biosynthesis"]},{name:["00906","Carotenoid biosynthesis"]},{name:["00905","Brassinosteroid biosynthesis"]},{name:["00981","Insect hormone biosynthesis"]},{name:["00908","Zeatin biosynthesis"]},{name:["00903","Limonene and pinene degradation"]},{name:["00281","Geraniol degradation"]},{name:["01052","Type I polyketide structures"]},{name:["00522","Biosynthesis of 12-, 14- and 16-membered macrolides"]},{name:["01051","Biosynthesis of ansamycins"]},{name:["01059","Biosynthesis of enediyne antibiotics"]},{name:["01056","Biosynthesis of type II polyketide backbone"]},{name:["01057","Biosynthesis of type II polyketide products"]},{name:["00253","Tetracycline biosynthesis"]},{name:["00523","Polyketide sugar unit biosynthesis"]},{name:["01054","Nonribosomal peptide structures"]},{name:["01053","Biosynthesis of siderophore group nonribosomal peptides"]},{name:["01055","Biosynthesis of vancomycin group antibiotics"]}]},{name:"Biosynthesis of other secondary metabolites",children:[{name:["00940","Phenylpropanoid biosynthesis"]},{name:["00945","Stilbenoid, diarylheptanoid and gingerol biosynthesis"]},{name:["00941","Flavonoid biosynthesis"]},{name:["00944","Flavone and flavonol biosynthesis"]},{name:["00942","Anthocyanin biosynthesis"]},{name:["00943","Isoflavonoid biosynthesis"]},{name:["00901","Indole alkaloid biosynthesis"]},{name:["00403","Indole diterpene alkaloid biosynthesis"]},{name:["00950","Isoquinoline alkaloid biosynthesis"]},{name:["00960","Tropane, piperidine and pyridine alkaloid biosynthesis"]},{name:["01058","Acridone alkaloid biosynthesis"]},{name:["00232","Caffeine metabolism"]},{name:["00965","Betalain biosynthesis"]},{name:["00966","Glucosinolate biosynthesis"]},{name:["00402","Benzoxazinoid biosynthesis"]},{name:["00311","Penicillin and cephalosporin biosynthesis"]},{name:["00332","Carbapenem biosynthesis"]},{name:["00261","Monobactam biosynthesis"]},{name:["00331","Clavulanic acid biosynthesis"]},{name:["00521","Streptomycin biosynthesis"]},{name:["00524","Neomycin, kanamycin and gentamicin biosynthesis"]},{name:["00525","Acarbose and validamycin biosynthesis"]},{name:["00401","Novobiocin biosynthesis"]},{name:["00404","Staurosporine biosynthesis"]},{name:["00405","Phenazine biosynthesis"]},{name:["00333","Prodigiosin biosynthesis"]},{name:["00254","Aflatoxin biosynthesis"]},{name:["00999","Biosynthesis of various secondary metabolites - part 1"]},{name:["00998","Biosynthesis of various secondary metabolites - part 2"]},{name:["00997","Biosynthesis of various secondary metabolites - part 3"]}]},{name:"Xenobiotics biodegradation and metabolism",children:[{name:["00362","Benzoate degradation"]},{name:["00627","Aminobenzoate degradation"]},{name:["00364","Fluorobenzoate degradation"]},{name:["00625","Chloroalkane and chloroalkene degradation"]},{name:["00361","Chlorocyclohexane and chlorobenzene degradation"]},{name:["00623","Toluene degradation"]},{name:["00622","Xylene degradation"]},{name:["00633","Nitrotoluene degradation"]},{name:["00642","Ethylbenzene degradation"]},{name:["00643","Styrene degradation"]},{name:["00791","Atrazine degradation"]},{name:["00930","Caprolactam degradation"]},{name:["00363","Bisphenol degradation"]},{name:["00621","Dioxin degradation"]},{name:["00626","Naphthalene degradation"]},{name:["00624","Polycyclic aromatic hydrocarbon degradation"]},{name:["00365","Furfural degradation"]},{name:["00984","Steroid degradation"]},{name:["00980","Metabolism of xenobiotics by cytochrome P450"]},{name:["00982","Drug metabolism - cytochrome P450"]},{name:["00983","Drug metabolism - other enzymes"]}]},{name:"Chemical structure transformation maps",children:[{name:["01010","Overview of biosynthetic pathways"]},{name:["01060","Biosynthesis of plant secondary metabolites"]},{name:["01061","Biosynthesis of phenylpropanoids"]},{name:["01062","Biosynthesis of terpenoids and steroids"]},{name:["01063","Biosynthesis of alkaloids derived from shikimate pathway"]},{name:["01064","Biosynthesis of alkaloids derived from ornithine, lysine and nicotinic acid"]},{name:["01065","Biosynthesis of alkaloids derived from histidine and purine"]},{name:["01066","Biosynthesis of alkaloids derived from terpenoid and polyketide"]},{name:["01070","Biosynthesis of plant hormones"]}]}]},{name:"Genetic Information Processing",children:[{name:"Transcription",children:[{name:["03020","RNA polymerase"]},{name:["03022","Basal transcription factors"]},{name:["03040","Spliceosome"]}]},{name:"Translation",children:[{name:["03010","Ribosome"]},{name:["00970","Aminoacyl-tRNA biosynthesis"]},{name:["03013","RNA transport"]},{name:["03015","mRNA surveillance pathway"]},{name:["03008","Ribosome biogenesis in eukaryotes"]}]},{name:"Folding, sorting and degradation",children:[{name:["03060","Protein export"]},{name:["04141","Protein processing in endoplasmic reticulum"]},{name:["04130","SNARE interactions in vesicular transport"]},{name:["04120","Ubiquitin mediated proteolysis"]},{name:["04122","Sulfur relay system"]},{name:["03050","Proteasome"]},{name:["03018","RNA degradation"]}]},{name:"Replication and repair",children:[{name:["03030","DNA replication"]},{name:["03410","Base excision repair"]},{name:["03420","Nucleotide excision repair"]},{name:["03430","Mismatch repair"]},{name:["03440","Homologous recombination"]},{name:["03450","Non-homologous end-joining"]},{name:["03460","Fanconi anemia pathway"]}]}]},{name:"Environmental Information Processing",children:[{name:"Membrane transport",children:[{name:["02010","ABC transporters"]},{name:["02060","Phosphotransferase system (PTS)"]},{name:["03070","Bacterial secretion system"]}]},{name:"Signal transduction",children:[{name:["02020","Two-component system"]},{name:["04010","MAPK signaling pathway"]},{name:["04013","MAPK signaling pathway - fly"]},{name:["04016","MAPK signaling pathway - plant"]},{name:["04011","MAPK signaling pathway - yeast"]},{name:["04012","ErbB signaling pathway"]},{name:["04014","Ras signaling pathway"]},{name:["04015","Rap1 signaling pathway"]},{name:["04310","Wnt signaling pathway"]},{name:["04330","Notch signaling pathway"]},{name:["04340","Hedgehog signaling pathway"]},{name:["04341","Hedgehog signaling pathway - fly"]},{name:["04350","TGF-beta signaling pathway"]},{name:["04390","Hippo signaling pathway"]},{name:["04391","Hippo signaling pathway - fly"]},{name:["04392","Hippo signaling pathway - multiple species"]},{name:["04370","VEGF signaling pathway"]},{name:["04371","Apelin signaling pathway"]},{name:["04630","JAK-STAT signaling pathway"]},{name:["04064","NF-kappa B signaling pathway"]},{name:["04668","TNF signaling pathway"]},{name:["04066","HIF-1 signaling pathway"]},{name:["04068","FoxO signaling pathway"]},{name:["04020","Calcium signaling pathway"]},{name:["04070","Phosphatidylinositol signaling system"]},{name:["04072","Phospholipase D signaling pathway"]},{name:["04071","Sphingolipid signaling pathway"]},{name:["04024","cAMP signaling pathway"]},{name:["04022","cGMP-PKG signaling pathway"]},{name:["04151","PI3K-Akt signaling pathway"]},{name:["04152","AMPK signaling pathway"]},{name:["04150","mTOR signaling pathway"]},{name:["04075","Plant hormone signal transduction"]}]},{name:"Signaling molecules and interaction",children:[{name:["04080","Neuroactive ligand-receptor interaction"]},{name:["04060","Cytokine-cytokine receptor interaction"]},{name:["04061","Viral protein interaction with cytokine and cytokine receptor"]},{name:["04512","ECM-receptor interaction"]},{name:["04514","Cell adhesion molecules"]}]}]},{name:"Cellular Processes",children:[{name:"Transport and catabolism",children:[{name:["04144","Endocytosis"]},{name:["04145","Phagosome"]},{name:["04142","Lysosome"]},{name:["04146","Peroxisome"]},{name:["04140","Autophagy - animal"]},{name:["04138","Autophagy - yeast"]},{name:["04136","Autophagy - other"]},{name:["04137","Mitophagy - animal"]},{name:["04139","Mitophagy - yeast"]}]},{name:"Cell growth and death",children:[{name:["04110","Cell cycle"]},{name:["04111","Cell cycle - yeast"]},{name:["04112","Cell cycle - Caulobacter"]},{name:["04113","Meiosis - yeast"]},{name:["04114","Oocyte meiosis"]},{name:["04210","Apoptosis"]},{name:["04214","Apoptosis - fly"]},{name:["04215","Apoptosis - multiple species"]},{name:["04216","Ferroptosis"]},{name:["04217","Necroptosis"]},{name:["04115","p53 signaling pathway"]},{name:["04218","Cellular senescence"]}]},{name:"Cellular community - eukaryotes",children:[{name:["04510","Focal adhesion"]},{name:["04520","Adherens junction"]},{name:["04530","Tight junction"]},{name:["04540","Gap junction"]},{name:["04550","Signaling pathways regulating pluripotency of stem cells"]}]},{name:"Cellular community - prokaryotes",children:[{name:["02024","Quorum sensing"]},{name:["05111","Biofilm formation - Vibrio cholerae"]},{name:["02025","Biofilm formation - Pseudomonas aeruginosa"]},{name:["02026","Biofilm formation - Escherichia coli"]}]},{name:"Cell motility",children:[{name:["02030","Bacterial chemotaxis"]},{name:["02040","Flagellar assembly"]},{name:["04810","Regulation of actin cytoskeleton"]}]}]},{name:"Organismal Systems",children:[{name:"Immune system",children:[{name:["04640","Hematopoietic cell lineage"]},{name:["04610","Complement and coagulation cascades"]},{name:["04611","Platelet activation"]},{name:["04620","Toll-like receptor signaling pathway"]},{name:["04624","Toll and Imd signaling pathway"]},{name:["04621","NOD-like receptor signaling pathway"]},{name:["04622","RIG-I-like receptor signaling pathway"]},{name:["04623","Cytosolic DNA-sensing pathway"]},{name:["04625","C-type lectin receptor signaling pathway"]},{name:["04650","Natural killer cell mediated cytotoxicity"]},{name:["04612","Antigen processing and presentation"]},{name:["04660","T cell receptor signaling pathway"]},{name:["04658","Th1 and Th2 cell differentiation"]},{name:["04659","Th17 cell differentiation"]},{name:["04657","IL-17 signaling pathway"]},{name:["04662","B cell receptor signaling pathway"]},{name:["04664","Fc epsilon RI signaling pathway"]},{name:["04666","Fc gamma R-mediated phagocytosis"]},{name:["04670","Leukocyte transendothelial migration"]},{name:["04672","Intestinal immune network for IgA production"]},{name:["04062","Chemokine signaling pathway"]}]},{name:"Endocrine system",children:[{name:["04911","Insulin secretion"]},{name:["04910","Insulin signaling pathway"]},{name:["04922","Glucagon signaling pathway"]},{name:["04923","Regulation of lipolysis in adipocytes"]},{name:["04920","Adipocytokine signaling pathway"]},{name:["03320","PPAR signaling pathway"]},{name:["04929","GnRH secretion"]},{name:["04912","GnRH signaling pathway"]},{name:["04913","Ovarian steroidogenesis"]},{name:["04915","Estrogen signaling pathway"]},{name:["04914","Progesterone-mediated oocyte maturation"]},{name:["04917","Prolactin signaling pathway"]},{name:["04921","Oxytocin signaling pathway"]},{name:["04926","Relaxin signaling pathway"]},{name:["04935","Growth hormone synthesis, secretion and action"]},{name:["04918","Thyroid hormone synthesis"]},{name:["04919","Thyroid hormone signaling pathway"]},{name:["04928","Parathyroid hormone synthesis, secretion and action"]},{name:["04916","Melanogenesis"]},{name:["04924","Renin secretion"]},{name:["04614","Renin-angiotensin system"]},{name:["04925","Aldosterone synthesis and secretion"]},{name:["04927","Cortisol synthesis and secretion"]}]},{name:"Circulatory system",children:[{name:["04260","Cardiac muscle contraction"]},{name:["04261","Adrenergic signaling in cardiomyocytes"]},{name:["04270","Vascular smooth muscle contraction"]}]},{name:"Digestive system",children:[{name:["04970","Salivary secretion"]},{name:["04971","Gastric acid secretion"]},{name:["04972","Pancreatic secretion"]},{name:["04976","Bile secretion"]},{name:["04973","Carbohydrate digestion and absorption"]},{name:["04974","Protein digestion and absorption"]},{name:["04975","Fat digestion and absorption"]},{name:["04979","Cholesterol metabolism"]},{name:["04977","Vitamin digestion and absorption"]},{name:["04978","Mineral absorption"]}]},{name:"Excretory system",children:[{name:["04962","Vasopressin-regulated water reabsorption"]},{name:["04960","Aldosterone-regulated sodium reabsorption"]},{name:["04961","Endocrine and other factor-regulated calcium reabsorption"]},{name:["04964","Proximal tubule bicarbonate reclamation"]},{name:["04966","Collecting duct acid secretion"]}]},{name:"Nervous system",children:[{name:["04724","Glutamatergic synapse"]},{name:["04727","GABAergic synapse"]},{name:["04725","Cholinergic synapse"]},{name:["04728","Dopaminergic synapse"]},{name:["04726","Serotonergic synapse"]},{name:["04720","Long-term potentiation"]},{name:["04730","Long-term depression"]},{name:["04723","Retrograde endocannabinoid signaling"]},{name:["04721","Synaptic vesicle cycle"]},{name:["04722","Neurotrophin signaling pathway"]}]},{name:"Sensory system",children:[{name:["04744","Phototransduction"]},{name:["04745","Phototransduction - fly"]},{name:["04740","Olfactory transduction"]},{name:["04742","Taste transduction"]},{name:["04750","Inflammatory mediator regulation of TRP channels"]}]},{name:"Development and regeneration",children:[{name:["04320","Dorso-ventral axis formation"]},{name:["04360","Axon guidance"]},{name:["04361","Axon regeneration"]},{name:["04380","Osteoclast differentiation"]}]},{name:"Aging",children:[{name:["04211","Longevity regulating pathway"]},{name:["04212","Longevity regulating pathway - worm"]},{name:["04213","Longevity regulating pathway - multiple species"]}]},{name:"Environmental adaptation",children:[{name:["04710","Circadian rhythm"]},{name:["04713","Circadian entrainment"]},{name:["04711","Circadian rhythm - fly"]},{name:["04712","Circadian rhythm - plant"]},{name:["04714","Thermogenesis"]},{name:["04626","Plant-pathogen interaction"]}]}]},{name:"Human Diseases",children:[{name:"Cancer: overview",children:[{name:["05200","Pathways in cancer"]},{name:["05202","Transcriptional misregulation in cancer"]},{name:["05206","MicroRNAs in cancer"]},{name:["05205","Proteoglycans in cancer"]},{name:["05204","Chemical carcinogenesis"]},{name:["05203","Viral carcinogenesis"]},{name:["05230","Central carbon metabolism in cancer"]},{name:["05231","Choline metabolism in cancer"]},{name:["05235","PD-L1 expression and PD-1 checkpoint pathway in cancer"]}]},{name:"Cancer: specific types",children:[{name:["05210","Colorectal cancer"]},{name:["05212","Pancreatic cancer"]},{name:["05225","Hepatocellular carcinoma"]},{name:["05226","Gastric cancer"]},{name:["05214","Glioma"]},{name:["05216","Thyroid cancer"]},{name:["05221","Acute myeloid leukemia"]},{name:["05220","Chronic myeloid leukemia"]},{name:["05217","Basal cell carcinoma"]},{name:["05218","Melanoma"]},{name:["05211","Renal cell carcinoma"]},{name:["05219","Bladder cancer"]},{name:["05215","Prostate cancer"]},{name:["05213","Endometrial cancer"]},{name:["05224","Breast cancer"]},{name:["05222","Small cell lung cancer"]},{name:["05223","Non-small cell lung cancer"]}]},{name:"Immune disease",children:[{name:["05310","Asthma"]},{name:["05322","Systemic lupus erythematosus"]},{name:["05323","Rheumatoid arthritis"]},{name:["05320","Autoimmune thyroid disease"]},{name:["05321","Inflammatory bowel disease"]},{name:["05330","Allograft rejection"]},{name:["05332","Graft-versus-host disease"]},{name:["05340","Primary immunodeficiency"]}]},{name:"Neurodegenerative disease",children:[{name:["05010","Alzheimer disease"]},{name:["05012","Parkinson disease"]},{name:["05014","Amyotrophic lateral sclerosis"]},{name:["05016","Huntington disease"]},{name:["05017","Spinocerebellar ataxia"]},{name:["05020","Prion disease"]},{name:["05022","Pathways of neurodegeneration - multiple diseases"]}]},{name:"Substance dependence",children:[{name:["05030","Cocaine addiction"]},{name:["05031","Amphetamine addiction"]},{name:["05032","Morphine addiction"]},{name:["05033","Nicotine addiction"]},{name:["05034","Alcoholism"]}]},{name:"Cardiovascular disease",children:[{name:["05418","Fluid shear stress and atherosclerosis"]},{name:["05410","Hypertrophic cardiomyopathy"]},{name:["05412","Arrhythmogenic right ventricular cardiomyopathy"]},{name:["05414","Dilated cardiomyopathy"]},{name:["05416","Viral myocarditis"]}]},{name:"Endocrine and metabolic disease",children:[{name:["04930","Type II diabetes mellitus"]},{name:["04940","Type I diabetes mellitus"]},{name:["04950","Maturity onset diabetes of the young"]},{name:["04932","Non-alcoholic fatty liver disease"]},{name:["04931","Insulin resistance"]},{name:["04933","AGE-RAGE signaling pathway in diabetic complications"]},{name:["04934","Cushing syndrome"]}]},{name:"Infectious disease: bacterial",children:[{name:["05110","Vibrio cholerae infection"]},{name:["05120","Epithelial cell signaling in Helicobacter pylori infection"]},{name:["05130","Pathogenic Escherichia coli infection"]},{name:["05132","Salmonella infection"]},{name:["05131","Shigellosis"]},{name:["05135","Yersinia infection"]},{name:["05133","Pertussis"]},{name:["05134","Legionellosis"]},{name:["05150","Staphylococcus aureus infection"]},{name:["05152","Tuberculosis"]},{name:["05100","Bacterial invasion of epithelial cells"]}]},{name:"Infectious disease: viral",children:[{name:["05166","Human T-cell leukemia virus 1 infection"]},{name:["05170","Human immunodeficiency virus 1 infection"]},{name:["05162","Measles"]},{name:["05164","Influenza A"]},{name:["05161","Hepatitis B"]},{name:["05160","Hepatitis C"]},{name:["05168","Herpes simplex virus 1 infection"]},{name:["05163","Human cytomegalovirus infection"]},{name:["05167","Kaposi sarcoma-associated herpesvirus infection"]},{name:["05169","Epstein-Barr virus infection"]},{name:["05165","Human papillomavirus infection"]}]},{name:"Infectious disease: parasitic",children:[{name:["05146","Amoebiasis"]},{name:["05144","Malaria"]},{name:["05145","Toxoplasmosis"]},{name:["05140","Leishmaniasis"]},{name:["05142","Chagas disease"]},{name:["05143","African trypanosomiasis"]}]},{name:"Drug resistance: antimicrobial",children:[{name:["01501","beta-Lactam resistance"]},{name:["01502","Vancomycin resistance"]},{name:["01503","Cationic antimicrobial peptide (CAMP) resistance"]}]},{name:"Drug resistance: antineoplastic",children:[{name:["01521","EGFR tyrosine kinase inhibitor resistance"]},{name:["01524","Platinum drug resistance"]},{name:["01523","Antifolate resistance"]},{name:["01522","Endocrine resistance"]}]}]},{name:"Drug Development",children:[{name:"Chronology: Antiinfectives",children:[{name:["07011","Penicillins"]},{name:["07012","Cephalosporins - parenteral agents"]},{name:["07013","Cephalosporins - oral agents"]},{name:["07021","Aminoglycosides"]},{name:["07019","Tetracyclines"]},{name:["07020","Macrolides and ketolides"]},{name:["07014","Quinolones"]},{name:["07023","Rifamycins"]},{name:["07026","Antifungal agents"]},{name:["07044","Antiviral agents"]},{name:["07053","Anti-HIV agents"]}]},{name:"Chronology: Antineoplastics",children:[{name:["07040","Antineoplastics - alkylating agents"]},{name:["07041","Antineoplastics - antimetabolic agents"]},{name:["07042","Antineoplastics - agents from natural products"]},{name:["07043","Antineoplastics - hormones"]},{name:["07045","Antineoplastics - protein kinase inhibitors"]}]},{name:"Chronology: Nervous system agents",children:[{name:["07032","Hypnotics"]},{name:["07030","Anxiolytics"]},{name:["07033","Anticonvulsants"]},{name:["07015","Local analgesics"]},{name:["07039","Opioid analgesics"]},{name:["07028","Antipsychotics"]},{name:["07029","Antipsychotics - phenothiazines"]},{name:["07031","Antipsychotics - butyrophenones"]},{name:["07027","Antidepressants"]},{name:["07056","Agents for Alzheimer-type dementia"]},{name:["07057","Antiparkinsonian agents"]}]},{name:"Chronology: Other drugs",children:[{name:["07055","Sulfonamide derivatives - overview"]},{name:["07016","Sulfonamide derivatives - sulfa drugs"]},{name:["07017","Sulfonamide derivatives - diuretics"]},{name:["07018","Sulfonamide derivatives - hypoglycemic agents"]},{name:["07037","Antiarrhythmic drugs"]},{name:["07038","Antiulcer drugs"]},{name:["07046","Immunosuppressive agents"]},{name:["07047","Osteoporosis drugs"]},{name:["07048","Antimigraines"]},{name:["07049","Antithrombosis agents"]},{name:["07050","Antirheumatics - DMARDs and biological agents"]},{name:["07051","Antidiabetics"]},{name:["07052","Antidyslipidemic agents"]},{name:["07054","Antiglaucoma agents"]}]},{name:"Target-based classification: G protein-coupled receptors",children:[{name:["07220","Cholinergic and anticholinergic drugs"]},{name:["07215","alpha-Adrenergic receptor agonists|antagonists"]},{name:["07214","beta-Adrenergic receptor agonists|antagonists"]},{name:["07213","Dopamine receptor agonists|antagonists"]},{name:["07212","Histamine H1 receptor antagonists"]},{name:["07227","Histamine H2|H3 receptor agonists|antagonists"]},{name:["07211","Serotonin receptor agonists|antagonists"]},{name:["07228","Eicosanoid receptor agonists|antagonists"]},{name:["07224","Opioid receptor agonists|antagonists"]},{name:["07229","Angiotensin receptor and endothelin receptor antagonists"]}]},{name:"Target-based classification: Nuclear receptors",children:[{name:["07225","Glucocorticoid and mineralocorticoid receptor agonists|antagonists"]},{name:["07226","Progesterone, androgen and estrogen receptor agonists|antagonists"]},{name:["07223","Retinoic acid receptor (RAR) and retinoid X receptor (RXR) agonists|antagonists"]},{name:["07222","Peroxisome proliferator-activated receptor (PPAR) agonists"]}]},{name:"Target-based classification: Ion channels",children:[{name:["07221","Nicotinic cholinergic receptor antagonists"]},{name:["07230","GABA-A receptor agonists|antagonists"]},{name:["07036","Calcium channel blocking drugs"]},{name:["07231","Sodium channel blocking drugs"]},{name:["07232","Potassium channel blocking and opening drugs"]},{name:["07235","N-Methyl-D-aspartic acid (NMDA) receptor antagonists"]}]},{name:"Target-based classification: Transporters",children:[{name:["07233","Ion transporter inhibitors"]},{name:["07234","Neurotransmitter transporter inhibitors"]}]},{name:"Target-based classification: Enzymes",children:[{name:["07216","Catecholamine transferase inhibitors"]},{name:["07219","Cyclooxygenase inhibitors"]},{name:["07024","HMG-CoA reductase inhibitors"]},{name:["07217","Renin-angiotensin system inhibitors"]},{name:["07218","HIV protease inhibitors"]}]},{name:"Structure-based classification",children:[{name:["07025","Quinolines"]},{name:["07034","Eicosanoids"]},{name:["07035","Prostaglandins"]}]},{name:"Skeleton-based classification",children:[{name:["07110","Benzoic acid family"]},{name:["07112","1,2-Diphenyl substitution family"]},{name:["07114","Naphthalene family"]},{name:["07117","Benzodiazepine family"]}]}]}]},Ze=function(e){var a=e.keggMapList,n=e.onChange;return Je.children.map((function(e,i){return Object(t.jsx)(Re,{title:e.name,children:e.children.map((function(e,i){return Object(t.jsx)(Re,{title:e.name,children:e.children.map((function(e,i){var o=e.name;return Object(t.jsx)(ce,{label:o[1],checked:a.indexOf(o[0])>-1,setChecked:function(){return function(e){var t=Object(Z.a)(a),i=t.indexOf(e);i>-1?t.splice(i,1):t.push(e),n(t)}(o[0])}},i)}))},i)}))},i)}))},$e=function(e){var a=e.configData,n=e.onConfigChange;return Object(t.jsx)("main",{className:"main",children:Object(t.jsxs)("card",{children:[Object(t.jsx)(y.a,{children:Object(t.jsx)(x.a,{variant:"h6",children:"KEGG metabolic maps for KEGGCharter"})}),Object(t.jsx)(Ze,{keggMapList:a.keggcharterMaps,onChange:function(e){return n("keggcharterMaps",e)}})]})})},ea=function(e){var a=e.configData,n=e.onConfigChange;return Object(t.jsx)(N,{children:Object(t.jsx)($e,{configData:a,onConfigChange:n})})},aa=function(e){var a=e.configData,n=e.onConfigChange;return Object(t.jsxs)(ae.a,{children:[Object(t.jsxs)(ne.a,{children:[Object(t.jsx)(de,{label:"Choose proteomics workflow",value:a.proteomicsWorkflow,onChange:function(e){return n("proteomicsWorkflow",e.target.value)},options:Oe}),Object(t.jsx)(ce,{label:"Use cRAP database",checked:a.useCrap,setChecked:function(e){return n("useCrap",e.target.checked)}}),a.useCrap?Object(t.jsx)(t.Fragment,{}):Object(t.jsx)(oe,{label:"Set the path of contaminants database",value:a.diamondDatabase,onChange:function(e){return n("proteomicsContaminantesDatabase",e.target.value)},placeholder:xe.proteomicsContaminantesDatabase}),Object(t.jsx)(de,{label:"Choose taxonomic level to download reference proteomes",value:a.referenceProteomesTaxaLevel,onChange:function(e){return n("referenceProteomesTaxaLevel",e.target.value)},options:je}),Object(t.jsx)(oe,{label:"Set protease to use",value:a.diamondDatabase,onChange:function(e){return n("diamondDatabase",e.target.value)},placeholder:xe.diamondDatabase}),Object(t.jsx)(de,{label:"Choose protease used in proteomics experiment",value:a.protease,onChange:function(e){return n("protease",e.target.value)},options:ye}),"File"!=a.protease?Object(t.jsx)(t.Fragment,{}):Object(t.jsx)(oe,{label:"Set the filename of protease FASTA file",value:a.proteaseFile,onChange:function(e){return n("proteaseFile",e.target.value)},placeholder:xe.proteaseFile})]}),Object(t.jsx)(te.a,{style:{display:"flex",justifyContent:"flex-end"}})]})},na=function(e){var a=e.configData,n=e.onConfigChange;return Object(t.jsxs)(N,{children:[Object(t.jsx)(y.a,{children:Object(t.jsx)(x.a,{variant:"h6",children:"Configurations for metaproteomics analysis with MOSCA"})}),Object(t.jsx)(aa,{configData:a,onConfigChange:n})]})},ta=function(e){var a=e.outputsFiles;return Object(t.jsxs)(N,{outputsFiles:a,children:[Object(t.jsx)(y.a,{children:Object(t.jsx)(x.a,{variant:"h6",children:"FastQC Reports"})}),function(e){var a=0;return e.map((function(e){a++;var n=URL.createObjectURL(e.blob);return Object(t.jsx)(Re,{title:e.name,children:Object(t.jsx)("iframe",{src:n,style:{width:"100%",height:"1000px"}})},"accordion_".concat(a))}))}(a.qcReports)]})},ia=n(37),oa=n.n(ia),sa=function(e){var a=e.outputsFolder,n=Object(i.useState)([]),o=Object(m.a)(n,2),s=o[0],r=o[1],c=[],l=function(e,n){E.parse(e,{download:!0,header:!0,complete:function(e){var t;e.data.pop(),t={fileContent:e.data,fileName:n},c.push(t),a.length===c.length&&r(c)}})};Object(i.useEffect)((function(){a.map((function(e){var a=URL.createObjectURL(e.blob);l(a,e.name)}))}),[]);var d=0;return Object(t.jsxs)("main",{className:"main",children:[Object(t.jsx)(y.a,{children:Object(t.jsx)(x.a,{variant:"h6",children:"Quality check reports from assembly"})}),s.map((function(e){return d++,Object(t.jsx)(Re,{title:e.fileName,children:Object(t.jsx)(oa.a,{style:{width:"100%",height:"100%"},pagination:!0,noHeader:!0,columns:(a=e.fileContent,Object.keys(a[0]).map((function(e){return{name:(a=e,a.charAt(0).toUpperCase()+a.slice(1)),selector:e,sortable:!0};var a}))),data:e.fileContent})},"accordion_".concat(d));var a}))]})},ra=function(e){var a=e.outputsFolder;return Object(t.jsx)(N,{children:Object(t.jsx)(sa,{outputsFolder:a})})},ca=function(e){var a=e.outputsFiles;return Object(t.jsxs)(N,{children:[Object(t.jsx)(y.a,{children:Object(t.jsx)(x.a,{variant:"h6",children:"Annotation Results"})}),function(e){var a=0;return e.map((function(e){a++;var n=URL.createObjectURL(e.blob);return Object(t.jsx)(Re,{title:e.name,children:Object(t.jsx)("iframe",{src:n,style:{width:"100%",height:"1000px"}})},"accordion_".concat(a))}))}(a)]})},la=(n(57),n(76)),ma=n.n(la),da=function(e){var a=e.outputsFiles;return Object(t.jsxs)(N,{children:[Object(t.jsx)(y.a,{children:Object(t.jsx)(x.a,{variant:"h6",children:"Differential Analysis Results"})}),function(e){var a=0;return e.map((function(e){a++;var n=URL.createObjectURL(e.blob);return Object(t.jsxs)("div",{children:[Object(t.jsx)("h1",{style:{textAlign:"center",fontWeight:"bold",marginBottom:"1cm"},children:e.name}),Object(t.jsx)(ma.a,{image:{src:n,style:{margin:"auto",marginBottom:"2cm",justifyContent:"center"},className:"img"}})]},"blob_".concat(a))}))}(a)]})},ua=function(e){var a=e.outputsFiles,n=a.map((function(e){return e.name})),o=Object(i.useState)(!1),s=Object(m.a)(o,2),r=s[0],c=s[1],l=Object(i.useState)([]),d=Object(m.a)(l,2),u=d[0],p=d[1],h=0==r?function(){var e=Object(i.useState)([]),a=Object(m.a)(e,2),o=a[0],s=a[1];return Object(t.jsxs)("div",{children:[Object(t.jsx)(j.a,{variant:"contained",color:"primary",component:"label",onClick:function(){var e;0==(e=o).length|e.length>6?e.length>6?alert("Choose less KEGGMaps to display for a better performance"):alert("Please select KEGGMaps to display"):(p(o),c(!0))},children:"Click to view the KEGGMaps"}),Object(t.jsx)(Re,{title:"KeggMaps Results",children:n.map((function(e){return Object(t.jsx)(ce,{label:e[1],checked:o.indexOf(e[1])>-1,setChecked:function(){return function(e){var a=Object(Z.a)(o),n=a.indexOf(e);n>-1?a.splice(n,1):a.push(e),s(a)}(e[1])}},e[0])}))})]})}:function(){for(var e=[],n=0;n<a.length;n++)u.includes(a[n].name[1])&&e.push({name:a[n].name[1],blob:a[n].blob});return Object(t.jsxs)("div",{children:[Object(t.jsx)(j.a,{variant:"contained",color:"primary",component:"label",onClick:function(){c(!1)},children:"Click to view the KEGGMaps"}),function(e){var a=0;return e.map((function(e){a++;var n=URL.createObjectURL(e.blob);return Object(t.jsxs)("div",{children:[Object(t.jsx)("h1",{style:{textAlign:"center",fontWeight:"bold",marginBottom:"1cm"},children:e.name}),Object(t.jsx)(ma.a,{image:{src:n,style:{margin:"auto",justifyContent:"center"},className:"img"}}),Object(t.jsx)("br",{}),Object(t.jsx)("hr",{}),Object(t.jsx)("br",{})]},"blob_".concat(a))}))}(e)]})};return Object(t.jsx)(t.Fragment,{children:Object(t.jsx)("div",{children:Object(t.jsx)(h,{})})})},pa=function(e){var a=e.outputsFiles;return Object(t.jsx)(N,{children:Object(t.jsx)(ua,{outputsFiles:a})})},ha=function(e){var a=e.outputsFolder,n=Object(i.useState)([]),o=Object(m.a)(n,2),s=o[0],r=o[1],c=[],l=function(e,n){E.parse(e,{download:!0,header:!0,complete:function(e){e.data.pop();var t,i=e.data.map((function(e){for(var a={},n=0,t=Object.entries(e);n<t.length;n++){var i=Object(m.a)(t[n],2),o=i[0],s=i[1];a[o.replaceAll(".","/").replaceAll("[","(").replaceAll("]",")").replaceAll(":","_")]=s}return a}));t={fileContent:i,fileName:n},c.push(t),a.length===c.length&&r(c)}})};Object(i.useEffect)((function(){a.map((function(e){var a=URL.createObjectURL(e.blob);l(a,e.name)}))}),[]);return Object(t.jsxs)("main",{className:"main",children:[Object(t.jsx)(y.a,{children:Object(t.jsx)(x.a,{variant:"h6",children:"Entry Reports"})}),function(e){return void 0!=e[0]?Object(t.jsx)(oa.a,{style:{width:"100%",height:"100%",margin:"auto"},title:e[0].fileName,pagination:!0,paginationRowsPerPageOptions:[10,20,30,40,50],noHeader:!0,columns:(a=e[0].fileContent,Object.keys(a[0]).map((function(e){return{name:(a=e,a.charAt(0).toUpperCase()+a.slice(1)),selector:e,sortable:!0};var a}))),data:e[0].fileContent}):Object(t.jsx)("div",{});var a}(s)]})},ba=function(e){var a=e.outputsFolder;return Object(t.jsx)(N,{children:Object(t.jsx)(ha,{style:{width:"100%",height:"100%"},outputsFolder:a})})},ga=function(e){var a=e.outputsFolder,n=Object(i.useState)([]),o=Object(m.a)(n,2),s=o[0],r=o[1],c=[],l=function(e,n){E.parse(e,{download:!0,header:!0,complete:function(e){e.data.pop();var t,i=e.data.map((function(e){for(var a={},n=0,t=Object.entries(e);n<t.length;n++){var i=Object(m.a)(t[n],2),o=i[0],s=i[1];a[o.replaceAll(".","/").replaceAll("[","(").replaceAll("]",")").replaceAll(":","_").replace("Unnamed_ 0","Sample N\xba")]=s}return a}));t={fileContent:i,fileName:n},c.push(t),a.length===c.length&&r(c)}})};Object(i.useEffect)((function(){a.map((function(e){var a=URL.createObjectURL(e.blob);l(a,e.name)}))}),[]);return Object(t.jsxs)("main",{className:"main",children:[Object(t.jsx)(y.a,{children:Object(t.jsx)(x.a,{variant:"h6",children:"General Reports"})}),function(e){return void 0!=e[0]?Object(t.jsx)(oa.a,{style:{width:"100%",height:"100%"},title:e[0].fileName,noHeader:!0,columns:(a=e[0].fileContent,Object.keys(a[0]).map((function(e){return{name:(a=e,a.charAt(0).toUpperCase()+a.slice(1)),selector:e,sortable:!0};var a}))),data:e[0].fileContent}):Object(t.jsx)("div",{});var a}(s)]})},fa=function(e){var a=e.outputsFolder;return Object(t.jsx)(N,{children:Object(t.jsx)(ga,{outputsFolder:a})})},Oa=function(e){var a=e.outputsFolder,n=Object(i.useState)([]),o=Object(m.a)(n,2),s=o[0],r=o[1],c=[],l=function(e,n){E.parse(e,{download:!0,header:!0,complete:function(e){e.data.pop();var t,i=e.data.map((function(e){for(var a={},n=0,t=Object.entries(e);n<t.length;n++){var i=Object(m.a)(t[n],2),o=i[0],s=i[1];a[o.replaceAll(".","/").replaceAll("[","(").replaceAll("]",")").replaceAll(":","_")]=s}return a}));t={fileContent:i,fileName:n},c.push(t),a.length===c.length&&r(c)}})};Object(i.useEffect)((function(){a.map((function(e){var a=URL.createObjectURL(e.blob);l(a,e.name)}))}),[]);return Object(t.jsxs)("main",{className:"main",children:[Object(t.jsx)(y.a,{children:Object(t.jsx)(x.a,{variant:"h6",children:"Protein Reports"})}),function(e){return console.log(e),void 0!=e[0]?Object(t.jsx)(oa.a,{style:{width:"100%",height:"100%"},title:e[0].fileName,pagination:!0,paginationRowsPerPageOptions:[10,20,30,40,50],noHeader:!0,columns:(a=e[0].fileContent,Object.keys(a[0]).map((function(e){return console.log(e),{name:(a=e,a.charAt(0).toUpperCase()+a.slice(1)),selector:e,sortable:!0};var a}))),data:e[0].fileContent}):Object(t.jsx)("div",{});var a}(s)]})},ja=function(e){var a=e.outputsFolder;return Object(t.jsx)(N,{children:Object(t.jsx)(Oa,{outputsFolder:a})})},ya=function(e){var a=e.configData,n=e.onConfigChange,i=e.onConfigOverwrite,o=e.experiments,s=e.setExperiments,r=e.nExperimentsRows,c=e.setExperimentsRows,l=e.outputsFiles,m=e.setOutputsFiles;return Object(t.jsx)(d.a,{children:Object(t.jsxs)(u.c,{children:[Object(t.jsx)(u.a,{path:"/MOSGUITO/general-configuration",children:Object(t.jsx)(Me,{configData:a,onConfigChange:n,onConfigOverwrite:i})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/experiments",children:Object(t.jsx)(Ye,{experiments:o,setExperiments:s,nExperimentsRows:r,setExperimentsRows:c})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/uniprot-columns",children:Object(t.jsx)(We,{uniprotList:a.uniprotColumns,onChange:function(e){return n("uniprotColumns",e)}})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/uniprot-databases",children:Object(t.jsx)(Xe,{uniprotList:a.uniprotDatabases,onChange:function(e){return n("uniprotDatabases",e)}})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/keggmaps",children:Object(t.jsx)(ea,{configData:a,onConfigChange:n})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/proteomics-configuration",children:Object(t.jsx)(na,{configData:a,onConfigChange:n})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/about"}),Object(t.jsx)(u.a,{path:"/MOSGUITO/members",children:Object(t.jsx)(W,{})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/project",children:Object(t.jsx)(w,{})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/about",children:Object(t.jsx)(_,{})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/another/teams",children:Object(t.jsx)(Q,{})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/results",children:Object(t.jsx)(D,{outputsFiles:l,setOutputsFiles:m,onConfigOverwrite:i,setExperiments:s,setExperimentsRows:c})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/load-results",children:Object(t.jsx)(D,{outputsFiles:l,setOutputsFiles:m,onConfigChange:n,setExperiments:s,setExperimentsRows:c})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/fastqc-reports",children:Object(t.jsx)(ta,{outputsFiles:l})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/assembly-qc",children:Object(t.jsx)(ra,{outputsFolder:l.asReports})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/annotation-results",children:Object(t.jsx)(ca,{outputsFiles:l.KronaPlots})}),Object(t.jsx)(u.a,{path:"/MOSFUITO/differential-analysis",children:Object(t.jsx)(da,{outputsFiles:l.Heatmaps})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/keggmaps-results",children:Object(t.jsx)(pa,{outputsFiles:l.KEGGMaps})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/entry-reports",children:Object(t.jsx)(ba,{outputsFolder:l.entryReport})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/general-reports",children:Object(t.jsx)(fa,{outputsFolder:l.generalReport})}),Object(t.jsx)(u.a,{path:"/MOSGUITO/protein-reports",children:Object(t.jsx)(ja,{outputsFolder:l.proteinReport})}),Object(t.jsx)(u.a,{path:"/MOSGUITO",children:Object(t.jsx)(J,{})})]})})},xa=function(){var e=Object(i.useState)([{Files:"",Sample:"","Data type":"",Condition:"",Name:""}]),a=Object(m.a)(e,2),n=a[0],o=a[1],s=Object(i.useState)(xe),r=Object(m.a)(s,2),d=r[0],u=r[1],p=Object(i.useState)(1),h=Object(m.a)(p,2),b=h[0],g=h[1],f=Object(i.useState)({}),O=Object(m.a)(f,2),j=O[0],y=O[1];return Object(t.jsx)(ya,{configData:d,onConfigChange:function(e,a){var n=Object(l.a)(Object(l.a)({},d),{},Object(c.a)({},e,a));u(n)},onConfigOverwrite:function(e){u(e)},nExperimentsRows:b,setExperimentsRows:g,experiments:n,setExperiments:o,outputsFiles:j,setOutputsFiles:y})},Ca=function(e){e&&e instanceof Function&&n.e(3).then(n.bind(null,289)).then((function(a){var n=a.getCLS,t=a.getFID,i=a.getFCP,o=a.getLCP,s=a.getTTFB;n(e),t(e),i(e),o(e),s(e)}))};r.a.render(Object(t.jsx)(o.a.StrictMode,{children:Object(t.jsx)(xa,{})}),document.getElementById("root")),Ca()}},[[224,1,2]]]);
//# sourceMappingURL=main.aadf157f.chunk.js.map